// hardhat-tracer/dist/src/colors.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.colorWarning = exports.colorIndexed = exports.colorNameTag = exports.colorSstore = exports.colorSload = exports.colorKey = exports.colorError = exports.colorEvent = exports.colorFunction = exports.colorContract = exports.colorLabel = void 0;
const chalk_1 = __importDefault(require("chalk"));
exports.colorLabel = chalk_1.default.white;
exports.colorContract = chalk_1.default.green;
exports.colorFunction = chalk_1.default.cyan;
exports.colorEvent = chalk_1.default.yellow;
exports.colorError = chalk_1.default.red;
exports.colorKey = chalk_1.default.magenta;
exports.colorSload = chalk_1.default.blueBright;
exports.colorSstore = chalk_1.default.redBright;
exports.colorNameTag = chalk_1.default.italic;
exports.colorIndexed = chalk_1.default.italic;
exports.colorWarning = chalk_1.default.yellow;
