// @injectivelabs/ts-types/dist/cjs/cosmos.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DevnetCosmosChainId = exports.TestnetCosmosChainId = exports.CosmosChainId = exports.StreamOperation = void 0;
var StreamOperation;
(function (StreamOperation) {
    StreamOperation["Insert"] = "insert";
    StreamOperation["Delete"] = "delete";
    StreamOperation["Replace"] = "replace";
    StreamOperation["Update"] = "update";
    StreamOperation["Invalidate"] = "invalidate";
})(StreamOperation = exports.StreamOperation || (exports.StreamOperation = {}));
var CosmosChainId;
(function (CosmosChainId) {
    CosmosChainId["Injective"] = "injective-1";
    CosmosChainId["Cosmoshub"] = "cosmoshub-4";
    CosmosChainId["Juno"] = "juno-1";
    CosmosChainId["Osmosis"] = "osmosis-1";
    CosmosChainId["Terra"] = "columbus-5";
    CosmosChainId["TerraUST"] = "columbus-5";
    CosmosChainId["Chihuahua"] = "chihuahua-1";
    CosmosChainId["Axelar"] = "axelar-dojo-1";
    CosmosChainId["Evmos"] = "evmos_9001-2";
    CosmosChainId["Persistence"] = "core-1";
    CosmosChainId["Secret"] = "secret-4";
    CosmosChainId["Stride"] = "stride-1";
    CosmosChainId["Crescent"] = "crescent-1";
    CosmosChainId["Sommelier"] = "sommelier-3";
    CosmosChainId["Canto"] = "canto_7700-1";
})(CosmosChainId = exports.CosmosChainId || (exports.CosmosChainId = {}));
var TestnetCosmosChainId;
(function (TestnetCosmosChainId) {
    TestnetCosmosChainId["Injective"] = "injective-888";
    TestnetCosmosChainId["Cosmoshub"] = "cosmoshub-testnet";
})(TestnetCosmosChainId = exports.TestnetCosmosChainId || (exports.TestnetCosmosChainId = {}));
var DevnetCosmosChainId;
(function (DevnetCosmosChainId) {
    DevnetCosmosChainId["Injective"] = "injective-777";
    DevnetCosmosChainId["Injective1"] = "injective-777";
})(DevnetCosmosChainId = exports.DevnetCosmosChainId || (exports.DevnetCosmosChainId = {}));
