// @injectivelabs/chain-api/injective/exchange/v1beta1/genesis_pb.js
// source: injective/exchange/v1beta1/genesis.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() {
  if (this) { return this; }
  if (typeof window !== 'undefined') { return window; }
  if (typeof global !== 'undefined') { return global; }
  if (typeof self !== 'undefined') { return self; }
  return Function('return this')();
}.call(null));

var injective_exchange_v1beta1_exchange_pb = require('../../../injective/exchange/v1beta1/exchange_pb.js');
goog.object.extend(proto, injective_exchange_v1beta1_exchange_pb);
var injective_exchange_v1beta1_tx_pb = require('../../../injective/exchange/v1beta1/tx_pb.js');
goog.object.extend(proto, injective_exchange_v1beta1_tx_pb);
var gogoproto_gogo_pb = require('../../../gogoproto/gogo_pb.js');
goog.object.extend(proto, gogoproto_gogo_pb);
goog.exportSymbol('proto.injective.exchange.v1beta1.AccountVolume', null, global);
goog.exportSymbol('proto.injective.exchange.v1beta1.Balance', null, global);
goog.exportSymbol('proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook', null, global);
goog.exportSymbol('proto.injective.exchange.v1beta1.DerivativeOrderBook', null, global);
goog.exportSymbol('proto.injective.exchange.v1beta1.DerivativePosition', null, global);
goog.exportSymbol('proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState', null, global);
goog.exportSymbol('proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL', null, global);
goog.exportSymbol('proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts', null, global);
goog.exportSymbol('proto.injective.exchange.v1beta1.GenesisState', null, global);
goog.exportSymbol('proto.injective.exchange.v1beta1.OrderbookSequence', null, global);
goog.exportSymbol('proto.injective.exchange.v1beta1.PerpetualMarketFundingState', null, global);
goog.exportSymbol('proto.injective.exchange.v1beta1.SpotOrderBook', null, global);
goog.exportSymbol('proto.injective.exchange.v1beta1.SubaccountNonce', null, global);
goog.exportSymbol('proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints', null, global);
goog.exportSymbol('proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.exchange.v1beta1.GenesisState = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.injective.exchange.v1beta1.GenesisState.repeatedFields_, null);
};
goog.inherits(proto.injective.exchange.v1beta1.GenesisState, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.exchange.v1beta1.GenesisState.displayName = 'proto.injective.exchange.v1beta1.GenesisState';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.exchange.v1beta1.OrderbookSequence = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.exchange.v1beta1.OrderbookSequence, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.exchange.v1beta1.OrderbookSequence.displayName = 'proto.injective.exchange.v1beta1.OrderbookSequence';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.displayName = 'proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.repeatedFields_, null);
};
goog.inherits(proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.displayName = 'proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.exchange.v1beta1.AccountVolume = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.exchange.v1beta1.AccountVolume, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.exchange.v1beta1.AccountVolume.displayName = 'proto.injective.exchange.v1beta1.AccountVolume';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.displayName = 'proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.repeatedFields_, null);
};
goog.inherits(proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.displayName = 'proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.exchange.v1beta1.SpotOrderBook = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.injective.exchange.v1beta1.SpotOrderBook.repeatedFields_, null);
};
goog.inherits(proto.injective.exchange.v1beta1.SpotOrderBook, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.exchange.v1beta1.SpotOrderBook.displayName = 'proto.injective.exchange.v1beta1.SpotOrderBook';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.exchange.v1beta1.DerivativeOrderBook = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.injective.exchange.v1beta1.DerivativeOrderBook.repeatedFields_, null);
};
goog.inherits(proto.injective.exchange.v1beta1.DerivativeOrderBook, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.exchange.v1beta1.DerivativeOrderBook.displayName = 'proto.injective.exchange.v1beta1.DerivativeOrderBook';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.repeatedFields_, null);
};
goog.inherits(proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.displayName = 'proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.exchange.v1beta1.Balance = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.exchange.v1beta1.Balance, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.exchange.v1beta1.Balance.displayName = 'proto.injective.exchange.v1beta1.Balance';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.exchange.v1beta1.DerivativePosition = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.exchange.v1beta1.DerivativePosition, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.exchange.v1beta1.DerivativePosition.displayName = 'proto.injective.exchange.v1beta1.DerivativePosition';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.exchange.v1beta1.SubaccountNonce = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.exchange.v1beta1.SubaccountNonce, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.exchange.v1beta1.SubaccountNonce.displayName = 'proto.injective.exchange.v1beta1.SubaccountNonce';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.displayName = 'proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.exchange.v1beta1.PerpetualMarketFundingState = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.exchange.v1beta1.PerpetualMarketFundingState, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.exchange.v1beta1.PerpetualMarketFundingState.displayName = 'proto.injective.exchange.v1beta1.PerpetualMarketFundingState';
}

/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.injective.exchange.v1beta1.GenesisState.repeatedFields_ = [2,3,4,5,6,7,8,9,10,11,12,16,17,19,20,22,23,24,25,26,27,28,29,30,31,32];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.exchange.v1beta1.GenesisState.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.exchange.v1beta1.GenesisState} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.GenesisState.toObject = function(includeInstance, msg) {
  var f, obj = {
    params: (f = msg.getParams()) && injective_exchange_v1beta1_exchange_pb.Params.toObject(includeInstance, f),
    spotMarketsList: jspb.Message.toObjectList(msg.getSpotMarketsList(),
    injective_exchange_v1beta1_exchange_pb.SpotMarket.toObject, includeInstance),
    derivativeMarketsList: jspb.Message.toObjectList(msg.getDerivativeMarketsList(),
    injective_exchange_v1beta1_exchange_pb.DerivativeMarket.toObject, includeInstance),
    spotOrderbookList: jspb.Message.toObjectList(msg.getSpotOrderbookList(),
    proto.injective.exchange.v1beta1.SpotOrderBook.toObject, includeInstance),
    derivativeOrderbookList: jspb.Message.toObjectList(msg.getDerivativeOrderbookList(),
    proto.injective.exchange.v1beta1.DerivativeOrderBook.toObject, includeInstance),
    balancesList: jspb.Message.toObjectList(msg.getBalancesList(),
    proto.injective.exchange.v1beta1.Balance.toObject, includeInstance),
    positionsList: jspb.Message.toObjectList(msg.getPositionsList(),
    proto.injective.exchange.v1beta1.DerivativePosition.toObject, includeInstance),
    subaccountTradeNoncesList: jspb.Message.toObjectList(msg.getSubaccountTradeNoncesList(),
    proto.injective.exchange.v1beta1.SubaccountNonce.toObject, includeInstance),
    expiryFuturesMarketInfoStateList: jspb.Message.toObjectList(msg.getExpiryFuturesMarketInfoStateList(),
    proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.toObject, includeInstance),
    perpetualMarketInfoList: jspb.Message.toObjectList(msg.getPerpetualMarketInfoList(),
    injective_exchange_v1beta1_exchange_pb.PerpetualMarketInfo.toObject, includeInstance),
    perpetualMarketFundingStateList: jspb.Message.toObjectList(msg.getPerpetualMarketFundingStateList(),
    proto.injective.exchange.v1beta1.PerpetualMarketFundingState.toObject, includeInstance),
    derivativeMarketSettlementScheduledList: jspb.Message.toObjectList(msg.getDerivativeMarketSettlementScheduledList(),
    injective_exchange_v1beta1_exchange_pb.DerivativeMarketSettlementInfo.toObject, includeInstance),
    isSpotExchangeEnabled: jspb.Message.getBooleanFieldWithDefault(msg, 13, false),
    isDerivativesExchangeEnabled: jspb.Message.getBooleanFieldWithDefault(msg, 14, false),
    tradingRewardCampaignInfo: (f = msg.getTradingRewardCampaignInfo()) && injective_exchange_v1beta1_exchange_pb.TradingRewardCampaignInfo.toObject(includeInstance, f),
    tradingRewardPoolCampaignScheduleList: jspb.Message.toObjectList(msg.getTradingRewardPoolCampaignScheduleList(),
    injective_exchange_v1beta1_exchange_pb.CampaignRewardPool.toObject, includeInstance),
    tradingRewardCampaignAccountPointsList: jspb.Message.toObjectList(msg.getTradingRewardCampaignAccountPointsList(),
    proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.toObject, includeInstance),
    feeDiscountSchedule: (f = msg.getFeeDiscountSchedule()) && injective_exchange_v1beta1_exchange_pb.FeeDiscountSchedule.toObject(includeInstance, f),
    feeDiscountAccountTierTtlList: jspb.Message.toObjectList(msg.getFeeDiscountAccountTierTtlList(),
    proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.toObject, includeInstance),
    feeDiscountBucketVolumeAccountsList: jspb.Message.toObjectList(msg.getFeeDiscountBucketVolumeAccountsList(),
    proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.toObject, includeInstance),
    isFirstFeeCycleFinished: jspb.Message.getBooleanFieldWithDefault(msg, 21, false),
    pendingTradingRewardPoolCampaignScheduleList: jspb.Message.toObjectList(msg.getPendingTradingRewardPoolCampaignScheduleList(),
    injective_exchange_v1beta1_exchange_pb.CampaignRewardPool.toObject, includeInstance),
    pendingTradingRewardCampaignAccountPointsList: jspb.Message.toObjectList(msg.getPendingTradingRewardCampaignAccountPointsList(),
    proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.toObject, includeInstance),
    rewardsOptOutAddressesList: (f = jspb.Message.getRepeatedField(msg, 24)) == null ? undefined : f,
    historicalTradeRecordsList: jspb.Message.toObjectList(msg.getHistoricalTradeRecordsList(),
    injective_exchange_v1beta1_exchange_pb.TradeRecords.toObject, includeInstance),
    binaryOptionsMarketsList: jspb.Message.toObjectList(msg.getBinaryOptionsMarketsList(),
    injective_exchange_v1beta1_exchange_pb.BinaryOptionsMarket.toObject, includeInstance),
    binaryOptionsMarketIdsScheduledForSettlementList: (f = jspb.Message.getRepeatedField(msg, 27)) == null ? undefined : f,
    spotMarketIdsScheduledToForceCloseList: (f = jspb.Message.getRepeatedField(msg, 28)) == null ? undefined : f,
    denomDecimalsList: jspb.Message.toObjectList(msg.getDenomDecimalsList(),
    injective_exchange_v1beta1_tx_pb.DenomDecimals.toObject, includeInstance),
    conditionalDerivativeOrderbooksList: jspb.Message.toObjectList(msg.getConditionalDerivativeOrderbooksList(),
    proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.toObject, includeInstance),
    marketFeeMultipliersList: jspb.Message.toObjectList(msg.getMarketFeeMultipliersList(),
    injective_exchange_v1beta1_exchange_pb.MarketFeeMultiplier.toObject, includeInstance),
    orderbookSequencesList: jspb.Message.toObjectList(msg.getOrderbookSequencesList(),
    proto.injective.exchange.v1beta1.OrderbookSequence.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.exchange.v1beta1.GenesisState}
 */
proto.injective.exchange.v1beta1.GenesisState.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.exchange.v1beta1.GenesisState;
  return proto.injective.exchange.v1beta1.GenesisState.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.exchange.v1beta1.GenesisState} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.exchange.v1beta1.GenesisState}
 */
proto.injective.exchange.v1beta1.GenesisState.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new injective_exchange_v1beta1_exchange_pb.Params;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.Params.deserializeBinaryFromReader);
      msg.setParams(value);
      break;
    case 2:
      var value = new injective_exchange_v1beta1_exchange_pb.SpotMarket;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.SpotMarket.deserializeBinaryFromReader);
      msg.addSpotMarkets(value);
      break;
    case 3:
      var value = new injective_exchange_v1beta1_exchange_pb.DerivativeMarket;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.DerivativeMarket.deserializeBinaryFromReader);
      msg.addDerivativeMarkets(value);
      break;
    case 4:
      var value = new proto.injective.exchange.v1beta1.SpotOrderBook;
      reader.readMessage(value,proto.injective.exchange.v1beta1.SpotOrderBook.deserializeBinaryFromReader);
      msg.addSpotOrderbook(value);
      break;
    case 5:
      var value = new proto.injective.exchange.v1beta1.DerivativeOrderBook;
      reader.readMessage(value,proto.injective.exchange.v1beta1.DerivativeOrderBook.deserializeBinaryFromReader);
      msg.addDerivativeOrderbook(value);
      break;
    case 6:
      var value = new proto.injective.exchange.v1beta1.Balance;
      reader.readMessage(value,proto.injective.exchange.v1beta1.Balance.deserializeBinaryFromReader);
      msg.addBalances(value);
      break;
    case 7:
      var value = new proto.injective.exchange.v1beta1.DerivativePosition;
      reader.readMessage(value,proto.injective.exchange.v1beta1.DerivativePosition.deserializeBinaryFromReader);
      msg.addPositions(value);
      break;
    case 8:
      var value = new proto.injective.exchange.v1beta1.SubaccountNonce;
      reader.readMessage(value,proto.injective.exchange.v1beta1.SubaccountNonce.deserializeBinaryFromReader);
      msg.addSubaccountTradeNonces(value);
      break;
    case 9:
      var value = new proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState;
      reader.readMessage(value,proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.deserializeBinaryFromReader);
      msg.addExpiryFuturesMarketInfoState(value);
      break;
    case 10:
      var value = new injective_exchange_v1beta1_exchange_pb.PerpetualMarketInfo;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.PerpetualMarketInfo.deserializeBinaryFromReader);
      msg.addPerpetualMarketInfo(value);
      break;
    case 11:
      var value = new proto.injective.exchange.v1beta1.PerpetualMarketFundingState;
      reader.readMessage(value,proto.injective.exchange.v1beta1.PerpetualMarketFundingState.deserializeBinaryFromReader);
      msg.addPerpetualMarketFundingState(value);
      break;
    case 12:
      var value = new injective_exchange_v1beta1_exchange_pb.DerivativeMarketSettlementInfo;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.DerivativeMarketSettlementInfo.deserializeBinaryFromReader);
      msg.addDerivativeMarketSettlementScheduled(value);
      break;
    case 13:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setIsSpotExchangeEnabled(value);
      break;
    case 14:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setIsDerivativesExchangeEnabled(value);
      break;
    case 15:
      var value = new injective_exchange_v1beta1_exchange_pb.TradingRewardCampaignInfo;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.TradingRewardCampaignInfo.deserializeBinaryFromReader);
      msg.setTradingRewardCampaignInfo(value);
      break;
    case 16:
      var value = new injective_exchange_v1beta1_exchange_pb.CampaignRewardPool;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.CampaignRewardPool.deserializeBinaryFromReader);
      msg.addTradingRewardPoolCampaignSchedule(value);
      break;
    case 17:
      var value = new proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints;
      reader.readMessage(value,proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.deserializeBinaryFromReader);
      msg.addTradingRewardCampaignAccountPoints(value);
      break;
    case 18:
      var value = new injective_exchange_v1beta1_exchange_pb.FeeDiscountSchedule;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.FeeDiscountSchedule.deserializeBinaryFromReader);
      msg.setFeeDiscountSchedule(value);
      break;
    case 19:
      var value = new proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL;
      reader.readMessage(value,proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.deserializeBinaryFromReader);
      msg.addFeeDiscountAccountTierTtl(value);
      break;
    case 20:
      var value = new proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts;
      reader.readMessage(value,proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.deserializeBinaryFromReader);
      msg.addFeeDiscountBucketVolumeAccounts(value);
      break;
    case 21:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setIsFirstFeeCycleFinished(value);
      break;
    case 22:
      var value = new injective_exchange_v1beta1_exchange_pb.CampaignRewardPool;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.CampaignRewardPool.deserializeBinaryFromReader);
      msg.addPendingTradingRewardPoolCampaignSchedule(value);
      break;
    case 23:
      var value = new proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints;
      reader.readMessage(value,proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.deserializeBinaryFromReader);
      msg.addPendingTradingRewardCampaignAccountPoints(value);
      break;
    case 24:
      var value = /** @type {string} */ (reader.readString());
      msg.addRewardsOptOutAddresses(value);
      break;
    case 25:
      var value = new injective_exchange_v1beta1_exchange_pb.TradeRecords;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.TradeRecords.deserializeBinaryFromReader);
      msg.addHistoricalTradeRecords(value);
      break;
    case 26:
      var value = new injective_exchange_v1beta1_exchange_pb.BinaryOptionsMarket;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.BinaryOptionsMarket.deserializeBinaryFromReader);
      msg.addBinaryOptionsMarkets(value);
      break;
    case 27:
      var value = /** @type {string} */ (reader.readString());
      msg.addBinaryOptionsMarketIdsScheduledForSettlement(value);
      break;
    case 28:
      var value = /** @type {string} */ (reader.readString());
      msg.addSpotMarketIdsScheduledToForceClose(value);
      break;
    case 29:
      var value = new injective_exchange_v1beta1_tx_pb.DenomDecimals;
      reader.readMessage(value,injective_exchange_v1beta1_tx_pb.DenomDecimals.deserializeBinaryFromReader);
      msg.addDenomDecimals(value);
      break;
    case 30:
      var value = new proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook;
      reader.readMessage(value,proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.deserializeBinaryFromReader);
      msg.addConditionalDerivativeOrderbooks(value);
      break;
    case 31:
      var value = new injective_exchange_v1beta1_exchange_pb.MarketFeeMultiplier;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.MarketFeeMultiplier.deserializeBinaryFromReader);
      msg.addMarketFeeMultipliers(value);
      break;
    case 32:
      var value = new proto.injective.exchange.v1beta1.OrderbookSequence;
      reader.readMessage(value,proto.injective.exchange.v1beta1.OrderbookSequence.deserializeBinaryFromReader);
      msg.addOrderbookSequences(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.exchange.v1beta1.GenesisState.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.exchange.v1beta1.GenesisState} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.GenesisState.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getParams();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      injective_exchange_v1beta1_exchange_pb.Params.serializeBinaryToWriter
    );
  }
  f = message.getSpotMarketsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      injective_exchange_v1beta1_exchange_pb.SpotMarket.serializeBinaryToWriter
    );
  }
  f = message.getDerivativeMarketsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      3,
      f,
      injective_exchange_v1beta1_exchange_pb.DerivativeMarket.serializeBinaryToWriter
    );
  }
  f = message.getSpotOrderbookList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      4,
      f,
      proto.injective.exchange.v1beta1.SpotOrderBook.serializeBinaryToWriter
    );
  }
  f = message.getDerivativeOrderbookList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      5,
      f,
      proto.injective.exchange.v1beta1.DerivativeOrderBook.serializeBinaryToWriter
    );
  }
  f = message.getBalancesList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      6,
      f,
      proto.injective.exchange.v1beta1.Balance.serializeBinaryToWriter
    );
  }
  f = message.getPositionsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      7,
      f,
      proto.injective.exchange.v1beta1.DerivativePosition.serializeBinaryToWriter
    );
  }
  f = message.getSubaccountTradeNoncesList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      8,
      f,
      proto.injective.exchange.v1beta1.SubaccountNonce.serializeBinaryToWriter
    );
  }
  f = message.getExpiryFuturesMarketInfoStateList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      9,
      f,
      proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.serializeBinaryToWriter
    );
  }
  f = message.getPerpetualMarketInfoList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      10,
      f,
      injective_exchange_v1beta1_exchange_pb.PerpetualMarketInfo.serializeBinaryToWriter
    );
  }
  f = message.getPerpetualMarketFundingStateList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      11,
      f,
      proto.injective.exchange.v1beta1.PerpetualMarketFundingState.serializeBinaryToWriter
    );
  }
  f = message.getDerivativeMarketSettlementScheduledList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      12,
      f,
      injective_exchange_v1beta1_exchange_pb.DerivativeMarketSettlementInfo.serializeBinaryToWriter
    );
  }
  f = message.getIsSpotExchangeEnabled();
  if (f) {
    writer.writeBool(
      13,
      f
    );
  }
  f = message.getIsDerivativesExchangeEnabled();
  if (f) {
    writer.writeBool(
      14,
      f
    );
  }
  f = message.getTradingRewardCampaignInfo();
  if (f != null) {
    writer.writeMessage(
      15,
      f,
      injective_exchange_v1beta1_exchange_pb.TradingRewardCampaignInfo.serializeBinaryToWriter
    );
  }
  f = message.getTradingRewardPoolCampaignScheduleList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      16,
      f,
      injective_exchange_v1beta1_exchange_pb.CampaignRewardPool.serializeBinaryToWriter
    );
  }
  f = message.getTradingRewardCampaignAccountPointsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      17,
      f,
      proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.serializeBinaryToWriter
    );
  }
  f = message.getFeeDiscountSchedule();
  if (f != null) {
    writer.writeMessage(
      18,
      f,
      injective_exchange_v1beta1_exchange_pb.FeeDiscountSchedule.serializeBinaryToWriter
    );
  }
  f = message.getFeeDiscountAccountTierTtlList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      19,
      f,
      proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.serializeBinaryToWriter
    );
  }
  f = message.getFeeDiscountBucketVolumeAccountsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      20,
      f,
      proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.serializeBinaryToWriter
    );
  }
  f = message.getIsFirstFeeCycleFinished();
  if (f) {
    writer.writeBool(
      21,
      f
    );
  }
  f = message.getPendingTradingRewardPoolCampaignScheduleList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      22,
      f,
      injective_exchange_v1beta1_exchange_pb.CampaignRewardPool.serializeBinaryToWriter
    );
  }
  f = message.getPendingTradingRewardCampaignAccountPointsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      23,
      f,
      proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.serializeBinaryToWriter
    );
  }
  f = message.getRewardsOptOutAddressesList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      24,
      f
    );
  }
  f = message.getHistoricalTradeRecordsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      25,
      f,
      injective_exchange_v1beta1_exchange_pb.TradeRecords.serializeBinaryToWriter
    );
  }
  f = message.getBinaryOptionsMarketsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      26,
      f,
      injective_exchange_v1beta1_exchange_pb.BinaryOptionsMarket.serializeBinaryToWriter
    );
  }
  f = message.getBinaryOptionsMarketIdsScheduledForSettlementList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      27,
      f
    );
  }
  f = message.getSpotMarketIdsScheduledToForceCloseList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      28,
      f
    );
  }
  f = message.getDenomDecimalsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      29,
      f,
      injective_exchange_v1beta1_tx_pb.DenomDecimals.serializeBinaryToWriter
    );
  }
  f = message.getConditionalDerivativeOrderbooksList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      30,
      f,
      proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.serializeBinaryToWriter
    );
  }
  f = message.getMarketFeeMultipliersList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      31,
      f,
      injective_exchange_v1beta1_exchange_pb.MarketFeeMultiplier.serializeBinaryToWriter
    );
  }
  f = message.getOrderbookSequencesList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      32,
      f,
      proto.injective.exchange.v1beta1.OrderbookSequence.serializeBinaryToWriter
    );
  }
};


/**
 * optional Params params = 1;
 * @return {?proto.injective.exchange.v1beta1.Params}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getParams = function() {
  return /** @type{?proto.injective.exchange.v1beta1.Params} */ (
    jspb.Message.getWrapperField(this, injective_exchange_v1beta1_exchange_pb.Params, 1));
};


/**
 * @param {?proto.injective.exchange.v1beta1.Params|undefined} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setParams = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearParams = function() {
  return this.setParams(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.hasParams = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * repeated SpotMarket spot_markets = 2;
 * @return {!Array<!proto.injective.exchange.v1beta1.SpotMarket>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getSpotMarketsList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.SpotMarket>} */ (
    jspb.Message.getRepeatedWrapperField(this, injective_exchange_v1beta1_exchange_pb.SpotMarket, 2));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.SpotMarket>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setSpotMarketsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.SpotMarket=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.SpotMarket}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addSpotMarkets = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.injective.exchange.v1beta1.SpotMarket, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearSpotMarketsList = function() {
  return this.setSpotMarketsList([]);
};


/**
 * repeated DerivativeMarket derivative_markets = 3;
 * @return {!Array<!proto.injective.exchange.v1beta1.DerivativeMarket>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getDerivativeMarketsList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.DerivativeMarket>} */ (
    jspb.Message.getRepeatedWrapperField(this, injective_exchange_v1beta1_exchange_pb.DerivativeMarket, 3));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.DerivativeMarket>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setDerivativeMarketsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 3, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.DerivativeMarket=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.DerivativeMarket}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addDerivativeMarkets = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.injective.exchange.v1beta1.DerivativeMarket, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearDerivativeMarketsList = function() {
  return this.setDerivativeMarketsList([]);
};


/**
 * repeated SpotOrderBook spot_orderbook = 4;
 * @return {!Array<!proto.injective.exchange.v1beta1.SpotOrderBook>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getSpotOrderbookList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.SpotOrderBook>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.injective.exchange.v1beta1.SpotOrderBook, 4));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.SpotOrderBook>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setSpotOrderbookList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 4, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.SpotOrderBook=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.SpotOrderBook}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addSpotOrderbook = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 4, opt_value, proto.injective.exchange.v1beta1.SpotOrderBook, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearSpotOrderbookList = function() {
  return this.setSpotOrderbookList([]);
};


/**
 * repeated DerivativeOrderBook derivative_orderbook = 5;
 * @return {!Array<!proto.injective.exchange.v1beta1.DerivativeOrderBook>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getDerivativeOrderbookList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.DerivativeOrderBook>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.injective.exchange.v1beta1.DerivativeOrderBook, 5));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.DerivativeOrderBook>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setDerivativeOrderbookList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 5, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.DerivativeOrderBook=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.DerivativeOrderBook}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addDerivativeOrderbook = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 5, opt_value, proto.injective.exchange.v1beta1.DerivativeOrderBook, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearDerivativeOrderbookList = function() {
  return this.setDerivativeOrderbookList([]);
};


/**
 * repeated Balance balances = 6;
 * @return {!Array<!proto.injective.exchange.v1beta1.Balance>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getBalancesList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.Balance>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.injective.exchange.v1beta1.Balance, 6));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.Balance>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setBalancesList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 6, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.Balance=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.Balance}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addBalances = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 6, opt_value, proto.injective.exchange.v1beta1.Balance, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearBalancesList = function() {
  return this.setBalancesList([]);
};


/**
 * repeated DerivativePosition positions = 7;
 * @return {!Array<!proto.injective.exchange.v1beta1.DerivativePosition>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getPositionsList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.DerivativePosition>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.injective.exchange.v1beta1.DerivativePosition, 7));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.DerivativePosition>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setPositionsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 7, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.DerivativePosition=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.DerivativePosition}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addPositions = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 7, opt_value, proto.injective.exchange.v1beta1.DerivativePosition, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearPositionsList = function() {
  return this.setPositionsList([]);
};


/**
 * repeated SubaccountNonce subaccount_trade_nonces = 8;
 * @return {!Array<!proto.injective.exchange.v1beta1.SubaccountNonce>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getSubaccountTradeNoncesList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.SubaccountNonce>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.injective.exchange.v1beta1.SubaccountNonce, 8));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.SubaccountNonce>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setSubaccountTradeNoncesList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 8, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.SubaccountNonce=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.SubaccountNonce}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addSubaccountTradeNonces = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 8, opt_value, proto.injective.exchange.v1beta1.SubaccountNonce, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearSubaccountTradeNoncesList = function() {
  return this.setSubaccountTradeNoncesList([]);
};


/**
 * repeated ExpiryFuturesMarketInfoState expiry_futures_market_info_state = 9;
 * @return {!Array<!proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getExpiryFuturesMarketInfoStateList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState, 9));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setExpiryFuturesMarketInfoStateList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 9, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addExpiryFuturesMarketInfoState = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 9, opt_value, proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearExpiryFuturesMarketInfoStateList = function() {
  return this.setExpiryFuturesMarketInfoStateList([]);
};


/**
 * repeated PerpetualMarketInfo perpetual_market_info = 10;
 * @return {!Array<!proto.injective.exchange.v1beta1.PerpetualMarketInfo>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getPerpetualMarketInfoList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.PerpetualMarketInfo>} */ (
    jspb.Message.getRepeatedWrapperField(this, injective_exchange_v1beta1_exchange_pb.PerpetualMarketInfo, 10));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.PerpetualMarketInfo>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setPerpetualMarketInfoList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 10, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.PerpetualMarketInfo=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.PerpetualMarketInfo}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addPerpetualMarketInfo = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 10, opt_value, proto.injective.exchange.v1beta1.PerpetualMarketInfo, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearPerpetualMarketInfoList = function() {
  return this.setPerpetualMarketInfoList([]);
};


/**
 * repeated PerpetualMarketFundingState perpetual_market_funding_state = 11;
 * @return {!Array<!proto.injective.exchange.v1beta1.PerpetualMarketFundingState>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getPerpetualMarketFundingStateList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.PerpetualMarketFundingState>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.injective.exchange.v1beta1.PerpetualMarketFundingState, 11));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.PerpetualMarketFundingState>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setPerpetualMarketFundingStateList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 11, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.PerpetualMarketFundingState=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.PerpetualMarketFundingState}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addPerpetualMarketFundingState = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 11, opt_value, proto.injective.exchange.v1beta1.PerpetualMarketFundingState, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearPerpetualMarketFundingStateList = function() {
  return this.setPerpetualMarketFundingStateList([]);
};


/**
 * repeated DerivativeMarketSettlementInfo derivative_market_settlement_scheduled = 12;
 * @return {!Array<!proto.injective.exchange.v1beta1.DerivativeMarketSettlementInfo>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getDerivativeMarketSettlementScheduledList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.DerivativeMarketSettlementInfo>} */ (
    jspb.Message.getRepeatedWrapperField(this, injective_exchange_v1beta1_exchange_pb.DerivativeMarketSettlementInfo, 12));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.DerivativeMarketSettlementInfo>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setDerivativeMarketSettlementScheduledList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 12, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.DerivativeMarketSettlementInfo=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.DerivativeMarketSettlementInfo}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addDerivativeMarketSettlementScheduled = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 12, opt_value, proto.injective.exchange.v1beta1.DerivativeMarketSettlementInfo, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearDerivativeMarketSettlementScheduledList = function() {
  return this.setDerivativeMarketSettlementScheduledList([]);
};


/**
 * optional bool is_spot_exchange_enabled = 13;
 * @return {boolean}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getIsSpotExchangeEnabled = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 13, false));
};


/**
 * @param {boolean} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.setIsSpotExchangeEnabled = function(value) {
  return jspb.Message.setProto3BooleanField(this, 13, value);
};


/**
 * optional bool is_derivatives_exchange_enabled = 14;
 * @return {boolean}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getIsDerivativesExchangeEnabled = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 14, false));
};


/**
 * @param {boolean} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.setIsDerivativesExchangeEnabled = function(value) {
  return jspb.Message.setProto3BooleanField(this, 14, value);
};


/**
 * optional TradingRewardCampaignInfo trading_reward_campaign_info = 15;
 * @return {?proto.injective.exchange.v1beta1.TradingRewardCampaignInfo}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getTradingRewardCampaignInfo = function() {
  return /** @type{?proto.injective.exchange.v1beta1.TradingRewardCampaignInfo} */ (
    jspb.Message.getWrapperField(this, injective_exchange_v1beta1_exchange_pb.TradingRewardCampaignInfo, 15));
};


/**
 * @param {?proto.injective.exchange.v1beta1.TradingRewardCampaignInfo|undefined} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setTradingRewardCampaignInfo = function(value) {
  return jspb.Message.setWrapperField(this, 15, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearTradingRewardCampaignInfo = function() {
  return this.setTradingRewardCampaignInfo(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.hasTradingRewardCampaignInfo = function() {
  return jspb.Message.getField(this, 15) != null;
};


/**
 * repeated CampaignRewardPool trading_reward_pool_campaign_schedule = 16;
 * @return {!Array<!proto.injective.exchange.v1beta1.CampaignRewardPool>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getTradingRewardPoolCampaignScheduleList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.CampaignRewardPool>} */ (
    jspb.Message.getRepeatedWrapperField(this, injective_exchange_v1beta1_exchange_pb.CampaignRewardPool, 16));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.CampaignRewardPool>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setTradingRewardPoolCampaignScheduleList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 16, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.CampaignRewardPool=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.CampaignRewardPool}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addTradingRewardPoolCampaignSchedule = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 16, opt_value, proto.injective.exchange.v1beta1.CampaignRewardPool, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearTradingRewardPoolCampaignScheduleList = function() {
  return this.setTradingRewardPoolCampaignScheduleList([]);
};


/**
 * repeated TradingRewardCampaignAccountPoints trading_reward_campaign_account_points = 17;
 * @return {!Array<!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getTradingRewardCampaignAccountPointsList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints, 17));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setTradingRewardCampaignAccountPointsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 17, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addTradingRewardCampaignAccountPoints = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 17, opt_value, proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearTradingRewardCampaignAccountPointsList = function() {
  return this.setTradingRewardCampaignAccountPointsList([]);
};


/**
 * optional FeeDiscountSchedule fee_discount_schedule = 18;
 * @return {?proto.injective.exchange.v1beta1.FeeDiscountSchedule}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getFeeDiscountSchedule = function() {
  return /** @type{?proto.injective.exchange.v1beta1.FeeDiscountSchedule} */ (
    jspb.Message.getWrapperField(this, injective_exchange_v1beta1_exchange_pb.FeeDiscountSchedule, 18));
};


/**
 * @param {?proto.injective.exchange.v1beta1.FeeDiscountSchedule|undefined} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setFeeDiscountSchedule = function(value) {
  return jspb.Message.setWrapperField(this, 18, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearFeeDiscountSchedule = function() {
  return this.setFeeDiscountSchedule(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.hasFeeDiscountSchedule = function() {
  return jspb.Message.getField(this, 18) != null;
};


/**
 * repeated FeeDiscountAccountTierTTL fee_discount_account_tier_ttl = 19;
 * @return {!Array<!proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getFeeDiscountAccountTierTtlList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL, 19));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setFeeDiscountAccountTierTtlList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 19, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addFeeDiscountAccountTierTtl = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 19, opt_value, proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearFeeDiscountAccountTierTtlList = function() {
  return this.setFeeDiscountAccountTierTtlList([]);
};


/**
 * repeated FeeDiscountBucketVolumeAccounts fee_discount_bucket_volume_accounts = 20;
 * @return {!Array<!proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getFeeDiscountBucketVolumeAccountsList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts, 20));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setFeeDiscountBucketVolumeAccountsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 20, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addFeeDiscountBucketVolumeAccounts = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 20, opt_value, proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearFeeDiscountBucketVolumeAccountsList = function() {
  return this.setFeeDiscountBucketVolumeAccountsList([]);
};


/**
 * optional bool is_first_fee_cycle_finished = 21;
 * @return {boolean}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getIsFirstFeeCycleFinished = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 21, false));
};


/**
 * @param {boolean} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.setIsFirstFeeCycleFinished = function(value) {
  return jspb.Message.setProto3BooleanField(this, 21, value);
};


/**
 * repeated CampaignRewardPool pending_trading_reward_pool_campaign_schedule = 22;
 * @return {!Array<!proto.injective.exchange.v1beta1.CampaignRewardPool>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getPendingTradingRewardPoolCampaignScheduleList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.CampaignRewardPool>} */ (
    jspb.Message.getRepeatedWrapperField(this, injective_exchange_v1beta1_exchange_pb.CampaignRewardPool, 22));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.CampaignRewardPool>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setPendingTradingRewardPoolCampaignScheduleList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 22, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.CampaignRewardPool=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.CampaignRewardPool}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addPendingTradingRewardPoolCampaignSchedule = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 22, opt_value, proto.injective.exchange.v1beta1.CampaignRewardPool, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearPendingTradingRewardPoolCampaignScheduleList = function() {
  return this.setPendingTradingRewardPoolCampaignScheduleList([]);
};


/**
 * repeated TradingRewardCampaignAccountPendingPoints pending_trading_reward_campaign_account_points = 23;
 * @return {!Array<!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getPendingTradingRewardCampaignAccountPointsList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints, 23));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setPendingTradingRewardCampaignAccountPointsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 23, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addPendingTradingRewardCampaignAccountPoints = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 23, opt_value, proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearPendingTradingRewardCampaignAccountPointsList = function() {
  return this.setPendingTradingRewardCampaignAccountPointsList([]);
};


/**
 * repeated string rewards_opt_out_addresses = 24;
 * @return {!Array<string>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getRewardsOptOutAddressesList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 24));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.setRewardsOptOutAddressesList = function(value) {
  return jspb.Message.setField(this, 24, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addRewardsOptOutAddresses = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 24, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearRewardsOptOutAddressesList = function() {
  return this.setRewardsOptOutAddressesList([]);
};


/**
 * repeated TradeRecords historical_trade_records = 25;
 * @return {!Array<!proto.injective.exchange.v1beta1.TradeRecords>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getHistoricalTradeRecordsList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.TradeRecords>} */ (
    jspb.Message.getRepeatedWrapperField(this, injective_exchange_v1beta1_exchange_pb.TradeRecords, 25));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.TradeRecords>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setHistoricalTradeRecordsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 25, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.TradeRecords=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.TradeRecords}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addHistoricalTradeRecords = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 25, opt_value, proto.injective.exchange.v1beta1.TradeRecords, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearHistoricalTradeRecordsList = function() {
  return this.setHistoricalTradeRecordsList([]);
};


/**
 * repeated BinaryOptionsMarket binary_options_markets = 26;
 * @return {!Array<!proto.injective.exchange.v1beta1.BinaryOptionsMarket>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getBinaryOptionsMarketsList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.BinaryOptionsMarket>} */ (
    jspb.Message.getRepeatedWrapperField(this, injective_exchange_v1beta1_exchange_pb.BinaryOptionsMarket, 26));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.BinaryOptionsMarket>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setBinaryOptionsMarketsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 26, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.BinaryOptionsMarket=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.BinaryOptionsMarket}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addBinaryOptionsMarkets = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 26, opt_value, proto.injective.exchange.v1beta1.BinaryOptionsMarket, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearBinaryOptionsMarketsList = function() {
  return this.setBinaryOptionsMarketsList([]);
};


/**
 * repeated string binary_options_market_ids_scheduled_for_settlement = 27;
 * @return {!Array<string>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getBinaryOptionsMarketIdsScheduledForSettlementList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 27));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.setBinaryOptionsMarketIdsScheduledForSettlementList = function(value) {
  return jspb.Message.setField(this, 27, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addBinaryOptionsMarketIdsScheduledForSettlement = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 27, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearBinaryOptionsMarketIdsScheduledForSettlementList = function() {
  return this.setBinaryOptionsMarketIdsScheduledForSettlementList([]);
};


/**
 * repeated string spot_market_ids_scheduled_to_force_close = 28;
 * @return {!Array<string>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getSpotMarketIdsScheduledToForceCloseList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 28));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.setSpotMarketIdsScheduledToForceCloseList = function(value) {
  return jspb.Message.setField(this, 28, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addSpotMarketIdsScheduledToForceClose = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 28, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearSpotMarketIdsScheduledToForceCloseList = function() {
  return this.setSpotMarketIdsScheduledToForceCloseList([]);
};


/**
 * repeated DenomDecimals denom_decimals = 29;
 * @return {!Array<!proto.injective.exchange.v1beta1.DenomDecimals>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getDenomDecimalsList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.DenomDecimals>} */ (
    jspb.Message.getRepeatedWrapperField(this, injective_exchange_v1beta1_tx_pb.DenomDecimals, 29));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.DenomDecimals>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setDenomDecimalsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 29, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.DenomDecimals=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.DenomDecimals}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addDenomDecimals = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 29, opt_value, proto.injective.exchange.v1beta1.DenomDecimals, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearDenomDecimalsList = function() {
  return this.setDenomDecimalsList([]);
};


/**
 * repeated ConditionalDerivativeOrderBook conditional_derivative_orderbooks = 30;
 * @return {!Array<!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getConditionalDerivativeOrderbooksList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook, 30));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setConditionalDerivativeOrderbooksList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 30, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addConditionalDerivativeOrderbooks = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 30, opt_value, proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearConditionalDerivativeOrderbooksList = function() {
  return this.setConditionalDerivativeOrderbooksList([]);
};


/**
 * repeated MarketFeeMultiplier market_fee_multipliers = 31;
 * @return {!Array<!proto.injective.exchange.v1beta1.MarketFeeMultiplier>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getMarketFeeMultipliersList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.MarketFeeMultiplier>} */ (
    jspb.Message.getRepeatedWrapperField(this, injective_exchange_v1beta1_exchange_pb.MarketFeeMultiplier, 31));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.MarketFeeMultiplier>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setMarketFeeMultipliersList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 31, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.MarketFeeMultiplier=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.MarketFeeMultiplier}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addMarketFeeMultipliers = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 31, opt_value, proto.injective.exchange.v1beta1.MarketFeeMultiplier, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearMarketFeeMultipliersList = function() {
  return this.setMarketFeeMultipliersList([]);
};


/**
 * repeated OrderbookSequence orderbook_sequences = 32;
 * @return {!Array<!proto.injective.exchange.v1beta1.OrderbookSequence>}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.getOrderbookSequencesList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.OrderbookSequence>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.injective.exchange.v1beta1.OrderbookSequence, 32));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.OrderbookSequence>} value
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
*/
proto.injective.exchange.v1beta1.GenesisState.prototype.setOrderbookSequencesList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 32, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.OrderbookSequence=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.OrderbookSequence}
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.addOrderbookSequences = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 32, opt_value, proto.injective.exchange.v1beta1.OrderbookSequence, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.GenesisState} returns this
 */
proto.injective.exchange.v1beta1.GenesisState.prototype.clearOrderbookSequencesList = function() {
  return this.setOrderbookSequencesList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.exchange.v1beta1.OrderbookSequence.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.exchange.v1beta1.OrderbookSequence.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.exchange.v1beta1.OrderbookSequence} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.OrderbookSequence.toObject = function(includeInstance, msg) {
  var f, obj = {
    sequence: jspb.Message.getFieldWithDefault(msg, 1, 0),
    marketId: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.exchange.v1beta1.OrderbookSequence}
 */
proto.injective.exchange.v1beta1.OrderbookSequence.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.exchange.v1beta1.OrderbookSequence;
  return proto.injective.exchange.v1beta1.OrderbookSequence.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.exchange.v1beta1.OrderbookSequence} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.exchange.v1beta1.OrderbookSequence}
 */
proto.injective.exchange.v1beta1.OrderbookSequence.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setSequence(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setMarketId(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.exchange.v1beta1.OrderbookSequence.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.exchange.v1beta1.OrderbookSequence.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.exchange.v1beta1.OrderbookSequence} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.OrderbookSequence.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getSequence();
  if (f !== 0) {
    writer.writeUint64(
      1,
      f
    );
  }
  f = message.getMarketId();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional uint64 sequence = 1;
 * @return {number}
 */
proto.injective.exchange.v1beta1.OrderbookSequence.prototype.getSequence = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.exchange.v1beta1.OrderbookSequence} returns this
 */
proto.injective.exchange.v1beta1.OrderbookSequence.prototype.setSequence = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional string market_id = 2;
 * @return {string}
 */
proto.injective.exchange.v1beta1.OrderbookSequence.prototype.getMarketId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.exchange.v1beta1.OrderbookSequence} returns this
 */
proto.injective.exchange.v1beta1.OrderbookSequence.prototype.setMarketId = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.toObject = function(includeInstance, msg) {
  var f, obj = {
    account: jspb.Message.getFieldWithDefault(msg, 1, ""),
    tierTtl: (f = msg.getTierTtl()) && injective_exchange_v1beta1_exchange_pb.FeeDiscountTierTTL.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL}
 */
proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL;
  return proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL}
 */
proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setAccount(value);
      break;
    case 2:
      var value = new injective_exchange_v1beta1_exchange_pb.FeeDiscountTierTTL;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.FeeDiscountTierTTL.deserializeBinaryFromReader);
      msg.setTierTtl(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getAccount();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getTierTtl();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      injective_exchange_v1beta1_exchange_pb.FeeDiscountTierTTL.serializeBinaryToWriter
    );
  }
};


/**
 * optional string account = 1;
 * @return {string}
 */
proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.prototype.getAccount = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL} returns this
 */
proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.prototype.setAccount = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional FeeDiscountTierTTL tier_ttl = 2;
 * @return {?proto.injective.exchange.v1beta1.FeeDiscountTierTTL}
 */
proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.prototype.getTierTtl = function() {
  return /** @type{?proto.injective.exchange.v1beta1.FeeDiscountTierTTL} */ (
    jspb.Message.getWrapperField(this, injective_exchange_v1beta1_exchange_pb.FeeDiscountTierTTL, 2));
};


/**
 * @param {?proto.injective.exchange.v1beta1.FeeDiscountTierTTL|undefined} value
 * @return {!proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL} returns this
*/
proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.prototype.setTierTtl = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL} returns this
 */
proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.prototype.clearTierTtl = function() {
  return this.setTierTtl(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.injective.exchange.v1beta1.FeeDiscountAccountTierTTL.prototype.hasTierTtl = function() {
  return jspb.Message.getField(this, 2) != null;
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.toObject = function(includeInstance, msg) {
  var f, obj = {
    bucketStartTimestamp: jspb.Message.getFieldWithDefault(msg, 1, 0),
    accountVolumeList: jspb.Message.toObjectList(msg.getAccountVolumeList(),
    proto.injective.exchange.v1beta1.AccountVolume.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts}
 */
proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts;
  return proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts}
 */
proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setBucketStartTimestamp(value);
      break;
    case 2:
      var value = new proto.injective.exchange.v1beta1.AccountVolume;
      reader.readMessage(value,proto.injective.exchange.v1beta1.AccountVolume.deserializeBinaryFromReader);
      msg.addAccountVolume(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getBucketStartTimestamp();
  if (f !== 0) {
    writer.writeInt64(
      1,
      f
    );
  }
  f = message.getAccountVolumeList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      proto.injective.exchange.v1beta1.AccountVolume.serializeBinaryToWriter
    );
  }
};


/**
 * optional int64 bucket_start_timestamp = 1;
 * @return {number}
 */
proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.prototype.getBucketStartTimestamp = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts} returns this
 */
proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.prototype.setBucketStartTimestamp = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * repeated AccountVolume account_volume = 2;
 * @return {!Array<!proto.injective.exchange.v1beta1.AccountVolume>}
 */
proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.prototype.getAccountVolumeList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.AccountVolume>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.injective.exchange.v1beta1.AccountVolume, 2));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.AccountVolume>} value
 * @return {!proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts} returns this
*/
proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.prototype.setAccountVolumeList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.AccountVolume=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.AccountVolume}
 */
proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.prototype.addAccountVolume = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.injective.exchange.v1beta1.AccountVolume, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts} returns this
 */
proto.injective.exchange.v1beta1.FeeDiscountBucketVolumeAccounts.prototype.clearAccountVolumeList = function() {
  return this.setAccountVolumeList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.exchange.v1beta1.AccountVolume.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.exchange.v1beta1.AccountVolume.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.exchange.v1beta1.AccountVolume} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.AccountVolume.toObject = function(includeInstance, msg) {
  var f, obj = {
    account: jspb.Message.getFieldWithDefault(msg, 1, ""),
    volume: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.exchange.v1beta1.AccountVolume}
 */
proto.injective.exchange.v1beta1.AccountVolume.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.exchange.v1beta1.AccountVolume;
  return proto.injective.exchange.v1beta1.AccountVolume.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.exchange.v1beta1.AccountVolume} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.exchange.v1beta1.AccountVolume}
 */
proto.injective.exchange.v1beta1.AccountVolume.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setAccount(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setVolume(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.exchange.v1beta1.AccountVolume.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.exchange.v1beta1.AccountVolume.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.exchange.v1beta1.AccountVolume} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.AccountVolume.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getAccount();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getVolume();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional string account = 1;
 * @return {string}
 */
proto.injective.exchange.v1beta1.AccountVolume.prototype.getAccount = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.exchange.v1beta1.AccountVolume} returns this
 */
proto.injective.exchange.v1beta1.AccountVolume.prototype.setAccount = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string volume = 2;
 * @return {string}
 */
proto.injective.exchange.v1beta1.AccountVolume.prototype.getVolume = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.exchange.v1beta1.AccountVolume} returns this
 */
proto.injective.exchange.v1beta1.AccountVolume.prototype.setVolume = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.toObject = function(includeInstance, msg) {
  var f, obj = {
    account: jspb.Message.getFieldWithDefault(msg, 1, ""),
    points: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints}
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints;
  return proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints}
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setAccount(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setPoints(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getAccount();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getPoints();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional string account = 1;
 * @return {string}
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.prototype.getAccount = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints} returns this
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.prototype.setAccount = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string points = 2;
 * @return {string}
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.prototype.getPoints = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints} returns this
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.prototype.setPoints = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.toObject = function(includeInstance, msg) {
  var f, obj = {
    rewardPoolStartTimestamp: jspb.Message.getFieldWithDefault(msg, 1, 0),
    accountPointsList: jspb.Message.toObjectList(msg.getAccountPointsList(),
    proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints}
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints;
  return proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints}
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setRewardPoolStartTimestamp(value);
      break;
    case 2:
      var value = new proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints;
      reader.readMessage(value,proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.deserializeBinaryFromReader);
      msg.addAccountPoints(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getRewardPoolStartTimestamp();
  if (f !== 0) {
    writer.writeInt64(
      1,
      f
    );
  }
  f = message.getAccountPointsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints.serializeBinaryToWriter
    );
  }
};


/**
 * optional int64 reward_pool_start_timestamp = 1;
 * @return {number}
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.prototype.getRewardPoolStartTimestamp = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints} returns this
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.prototype.setRewardPoolStartTimestamp = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * repeated TradingRewardCampaignAccountPoints account_points = 2;
 * @return {!Array<!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints>}
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.prototype.getAccountPointsList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints, 2));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints>} value
 * @return {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints} returns this
*/
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.prototype.setAccountPointsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints}
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.prototype.addAccountPoints = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPoints, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints} returns this
 */
proto.injective.exchange.v1beta1.TradingRewardCampaignAccountPendingPoints.prototype.clearAccountPointsList = function() {
  return this.setAccountPointsList([]);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.injective.exchange.v1beta1.SpotOrderBook.repeatedFields_ = [3];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.exchange.v1beta1.SpotOrderBook.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.exchange.v1beta1.SpotOrderBook.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.exchange.v1beta1.SpotOrderBook} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.SpotOrderBook.toObject = function(includeInstance, msg) {
  var f, obj = {
    marketId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    isbuyside: jspb.Message.getBooleanFieldWithDefault(msg, 2, false),
    ordersList: jspb.Message.toObjectList(msg.getOrdersList(),
    injective_exchange_v1beta1_exchange_pb.SpotLimitOrder.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.exchange.v1beta1.SpotOrderBook}
 */
proto.injective.exchange.v1beta1.SpotOrderBook.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.exchange.v1beta1.SpotOrderBook;
  return proto.injective.exchange.v1beta1.SpotOrderBook.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.exchange.v1beta1.SpotOrderBook} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.exchange.v1beta1.SpotOrderBook}
 */
proto.injective.exchange.v1beta1.SpotOrderBook.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setMarketId(value);
      break;
    case 2:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setIsbuyside(value);
      break;
    case 3:
      var value = new injective_exchange_v1beta1_exchange_pb.SpotLimitOrder;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.SpotLimitOrder.deserializeBinaryFromReader);
      msg.addOrders(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.exchange.v1beta1.SpotOrderBook.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.exchange.v1beta1.SpotOrderBook.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.exchange.v1beta1.SpotOrderBook} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.SpotOrderBook.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getMarketId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getIsbuyside();
  if (f) {
    writer.writeBool(
      2,
      f
    );
  }
  f = message.getOrdersList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      3,
      f,
      injective_exchange_v1beta1_exchange_pb.SpotLimitOrder.serializeBinaryToWriter
    );
  }
};


/**
 * optional string market_id = 1;
 * @return {string}
 */
proto.injective.exchange.v1beta1.SpotOrderBook.prototype.getMarketId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.exchange.v1beta1.SpotOrderBook} returns this
 */
proto.injective.exchange.v1beta1.SpotOrderBook.prototype.setMarketId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional bool isBuySide = 2;
 * @return {boolean}
 */
proto.injective.exchange.v1beta1.SpotOrderBook.prototype.getIsbuyside = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 2, false));
};


/**
 * @param {boolean} value
 * @return {!proto.injective.exchange.v1beta1.SpotOrderBook} returns this
 */
proto.injective.exchange.v1beta1.SpotOrderBook.prototype.setIsbuyside = function(value) {
  return jspb.Message.setProto3BooleanField(this, 2, value);
};


/**
 * repeated SpotLimitOrder orders = 3;
 * @return {!Array<!proto.injective.exchange.v1beta1.SpotLimitOrder>}
 */
proto.injective.exchange.v1beta1.SpotOrderBook.prototype.getOrdersList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.SpotLimitOrder>} */ (
    jspb.Message.getRepeatedWrapperField(this, injective_exchange_v1beta1_exchange_pb.SpotLimitOrder, 3));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.SpotLimitOrder>} value
 * @return {!proto.injective.exchange.v1beta1.SpotOrderBook} returns this
*/
proto.injective.exchange.v1beta1.SpotOrderBook.prototype.setOrdersList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 3, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.SpotLimitOrder=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.SpotLimitOrder}
 */
proto.injective.exchange.v1beta1.SpotOrderBook.prototype.addOrders = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.injective.exchange.v1beta1.SpotLimitOrder, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.SpotOrderBook} returns this
 */
proto.injective.exchange.v1beta1.SpotOrderBook.prototype.clearOrdersList = function() {
  return this.setOrdersList([]);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.injective.exchange.v1beta1.DerivativeOrderBook.repeatedFields_ = [3];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.exchange.v1beta1.DerivativeOrderBook.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.exchange.v1beta1.DerivativeOrderBook.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.exchange.v1beta1.DerivativeOrderBook} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.DerivativeOrderBook.toObject = function(includeInstance, msg) {
  var f, obj = {
    marketId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    isbuyside: jspb.Message.getBooleanFieldWithDefault(msg, 2, false),
    ordersList: jspb.Message.toObjectList(msg.getOrdersList(),
    injective_exchange_v1beta1_exchange_pb.DerivativeLimitOrder.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.exchange.v1beta1.DerivativeOrderBook}
 */
proto.injective.exchange.v1beta1.DerivativeOrderBook.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.exchange.v1beta1.DerivativeOrderBook;
  return proto.injective.exchange.v1beta1.DerivativeOrderBook.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.exchange.v1beta1.DerivativeOrderBook} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.exchange.v1beta1.DerivativeOrderBook}
 */
proto.injective.exchange.v1beta1.DerivativeOrderBook.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setMarketId(value);
      break;
    case 2:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setIsbuyside(value);
      break;
    case 3:
      var value = new injective_exchange_v1beta1_exchange_pb.DerivativeLimitOrder;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.DerivativeLimitOrder.deserializeBinaryFromReader);
      msg.addOrders(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.exchange.v1beta1.DerivativeOrderBook.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.exchange.v1beta1.DerivativeOrderBook.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.exchange.v1beta1.DerivativeOrderBook} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.DerivativeOrderBook.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getMarketId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getIsbuyside();
  if (f) {
    writer.writeBool(
      2,
      f
    );
  }
  f = message.getOrdersList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      3,
      f,
      injective_exchange_v1beta1_exchange_pb.DerivativeLimitOrder.serializeBinaryToWriter
    );
  }
};


/**
 * optional string market_id = 1;
 * @return {string}
 */
proto.injective.exchange.v1beta1.DerivativeOrderBook.prototype.getMarketId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.exchange.v1beta1.DerivativeOrderBook} returns this
 */
proto.injective.exchange.v1beta1.DerivativeOrderBook.prototype.setMarketId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional bool isBuySide = 2;
 * @return {boolean}
 */
proto.injective.exchange.v1beta1.DerivativeOrderBook.prototype.getIsbuyside = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 2, false));
};


/**
 * @param {boolean} value
 * @return {!proto.injective.exchange.v1beta1.DerivativeOrderBook} returns this
 */
proto.injective.exchange.v1beta1.DerivativeOrderBook.prototype.setIsbuyside = function(value) {
  return jspb.Message.setProto3BooleanField(this, 2, value);
};


/**
 * repeated DerivativeLimitOrder orders = 3;
 * @return {!Array<!proto.injective.exchange.v1beta1.DerivativeLimitOrder>}
 */
proto.injective.exchange.v1beta1.DerivativeOrderBook.prototype.getOrdersList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.DerivativeLimitOrder>} */ (
    jspb.Message.getRepeatedWrapperField(this, injective_exchange_v1beta1_exchange_pb.DerivativeLimitOrder, 3));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.DerivativeLimitOrder>} value
 * @return {!proto.injective.exchange.v1beta1.DerivativeOrderBook} returns this
*/
proto.injective.exchange.v1beta1.DerivativeOrderBook.prototype.setOrdersList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 3, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.DerivativeLimitOrder=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.DerivativeLimitOrder}
 */
proto.injective.exchange.v1beta1.DerivativeOrderBook.prototype.addOrders = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.injective.exchange.v1beta1.DerivativeLimitOrder, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.DerivativeOrderBook} returns this
 */
proto.injective.exchange.v1beta1.DerivativeOrderBook.prototype.clearOrdersList = function() {
  return this.setOrdersList([]);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.repeatedFields_ = [2,3,4,5];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.toObject = function(includeInstance, msg) {
  var f, obj = {
    marketId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    limitBuyOrdersList: jspb.Message.toObjectList(msg.getLimitBuyOrdersList(),
    injective_exchange_v1beta1_exchange_pb.DerivativeLimitOrder.toObject, includeInstance),
    marketBuyOrdersList: jspb.Message.toObjectList(msg.getMarketBuyOrdersList(),
    injective_exchange_v1beta1_exchange_pb.DerivativeMarketOrder.toObject, includeInstance),
    limitSellOrdersList: jspb.Message.toObjectList(msg.getLimitSellOrdersList(),
    injective_exchange_v1beta1_exchange_pb.DerivativeLimitOrder.toObject, includeInstance),
    marketSellOrdersList: jspb.Message.toObjectList(msg.getMarketSellOrdersList(),
    injective_exchange_v1beta1_exchange_pb.DerivativeMarketOrder.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook}
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook;
  return proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook}
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setMarketId(value);
      break;
    case 2:
      var value = new injective_exchange_v1beta1_exchange_pb.DerivativeLimitOrder;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.DerivativeLimitOrder.deserializeBinaryFromReader);
      msg.addLimitBuyOrders(value);
      break;
    case 3:
      var value = new injective_exchange_v1beta1_exchange_pb.DerivativeMarketOrder;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.DerivativeMarketOrder.deserializeBinaryFromReader);
      msg.addMarketBuyOrders(value);
      break;
    case 4:
      var value = new injective_exchange_v1beta1_exchange_pb.DerivativeLimitOrder;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.DerivativeLimitOrder.deserializeBinaryFromReader);
      msg.addLimitSellOrders(value);
      break;
    case 5:
      var value = new injective_exchange_v1beta1_exchange_pb.DerivativeMarketOrder;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.DerivativeMarketOrder.deserializeBinaryFromReader);
      msg.addMarketSellOrders(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getMarketId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getLimitBuyOrdersList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      injective_exchange_v1beta1_exchange_pb.DerivativeLimitOrder.serializeBinaryToWriter
    );
  }
  f = message.getMarketBuyOrdersList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      3,
      f,
      injective_exchange_v1beta1_exchange_pb.DerivativeMarketOrder.serializeBinaryToWriter
    );
  }
  f = message.getLimitSellOrdersList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      4,
      f,
      injective_exchange_v1beta1_exchange_pb.DerivativeLimitOrder.serializeBinaryToWriter
    );
  }
  f = message.getMarketSellOrdersList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      5,
      f,
      injective_exchange_v1beta1_exchange_pb.DerivativeMarketOrder.serializeBinaryToWriter
    );
  }
};


/**
 * optional string market_id = 1;
 * @return {string}
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.getMarketId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook} returns this
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.setMarketId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * repeated DerivativeLimitOrder limit_buy_orders = 2;
 * @return {!Array<!proto.injective.exchange.v1beta1.DerivativeLimitOrder>}
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.getLimitBuyOrdersList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.DerivativeLimitOrder>} */ (
    jspb.Message.getRepeatedWrapperField(this, injective_exchange_v1beta1_exchange_pb.DerivativeLimitOrder, 2));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.DerivativeLimitOrder>} value
 * @return {!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook} returns this
*/
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.setLimitBuyOrdersList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.DerivativeLimitOrder=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.DerivativeLimitOrder}
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.addLimitBuyOrders = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.injective.exchange.v1beta1.DerivativeLimitOrder, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook} returns this
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.clearLimitBuyOrdersList = function() {
  return this.setLimitBuyOrdersList([]);
};


/**
 * repeated DerivativeMarketOrder market_buy_orders = 3;
 * @return {!Array<!proto.injective.exchange.v1beta1.DerivativeMarketOrder>}
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.getMarketBuyOrdersList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.DerivativeMarketOrder>} */ (
    jspb.Message.getRepeatedWrapperField(this, injective_exchange_v1beta1_exchange_pb.DerivativeMarketOrder, 3));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.DerivativeMarketOrder>} value
 * @return {!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook} returns this
*/
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.setMarketBuyOrdersList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 3, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.DerivativeMarketOrder=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.DerivativeMarketOrder}
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.addMarketBuyOrders = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.injective.exchange.v1beta1.DerivativeMarketOrder, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook} returns this
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.clearMarketBuyOrdersList = function() {
  return this.setMarketBuyOrdersList([]);
};


/**
 * repeated DerivativeLimitOrder limit_sell_orders = 4;
 * @return {!Array<!proto.injective.exchange.v1beta1.DerivativeLimitOrder>}
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.getLimitSellOrdersList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.DerivativeLimitOrder>} */ (
    jspb.Message.getRepeatedWrapperField(this, injective_exchange_v1beta1_exchange_pb.DerivativeLimitOrder, 4));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.DerivativeLimitOrder>} value
 * @return {!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook} returns this
*/
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.setLimitSellOrdersList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 4, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.DerivativeLimitOrder=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.DerivativeLimitOrder}
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.addLimitSellOrders = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 4, opt_value, proto.injective.exchange.v1beta1.DerivativeLimitOrder, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook} returns this
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.clearLimitSellOrdersList = function() {
  return this.setLimitSellOrdersList([]);
};


/**
 * repeated DerivativeMarketOrder market_sell_orders = 5;
 * @return {!Array<!proto.injective.exchange.v1beta1.DerivativeMarketOrder>}
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.getMarketSellOrdersList = function() {
  return /** @type{!Array<!proto.injective.exchange.v1beta1.DerivativeMarketOrder>} */ (
    jspb.Message.getRepeatedWrapperField(this, injective_exchange_v1beta1_exchange_pb.DerivativeMarketOrder, 5));
};


/**
 * @param {!Array<!proto.injective.exchange.v1beta1.DerivativeMarketOrder>} value
 * @return {!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook} returns this
*/
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.setMarketSellOrdersList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 5, value);
};


/**
 * @param {!proto.injective.exchange.v1beta1.DerivativeMarketOrder=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.exchange.v1beta1.DerivativeMarketOrder}
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.addMarketSellOrders = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 5, opt_value, proto.injective.exchange.v1beta1.DerivativeMarketOrder, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook} returns this
 */
proto.injective.exchange.v1beta1.ConditionalDerivativeOrderBook.prototype.clearMarketSellOrdersList = function() {
  return this.setMarketSellOrdersList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.exchange.v1beta1.Balance.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.exchange.v1beta1.Balance.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.exchange.v1beta1.Balance} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.Balance.toObject = function(includeInstance, msg) {
  var f, obj = {
    subaccountId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    denom: jspb.Message.getFieldWithDefault(msg, 2, ""),
    deposits: (f = msg.getDeposits()) && injective_exchange_v1beta1_exchange_pb.Deposit.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.exchange.v1beta1.Balance}
 */
proto.injective.exchange.v1beta1.Balance.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.exchange.v1beta1.Balance;
  return proto.injective.exchange.v1beta1.Balance.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.exchange.v1beta1.Balance} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.exchange.v1beta1.Balance}
 */
proto.injective.exchange.v1beta1.Balance.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setSubaccountId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setDenom(value);
      break;
    case 3:
      var value = new injective_exchange_v1beta1_exchange_pb.Deposit;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.Deposit.deserializeBinaryFromReader);
      msg.setDeposits(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.exchange.v1beta1.Balance.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.exchange.v1beta1.Balance.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.exchange.v1beta1.Balance} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.Balance.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getSubaccountId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getDenom();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getDeposits();
  if (f != null) {
    writer.writeMessage(
      3,
      f,
      injective_exchange_v1beta1_exchange_pb.Deposit.serializeBinaryToWriter
    );
  }
};


/**
 * optional string subaccount_id = 1;
 * @return {string}
 */
proto.injective.exchange.v1beta1.Balance.prototype.getSubaccountId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.exchange.v1beta1.Balance} returns this
 */
proto.injective.exchange.v1beta1.Balance.prototype.setSubaccountId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string denom = 2;
 * @return {string}
 */
proto.injective.exchange.v1beta1.Balance.prototype.getDenom = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.exchange.v1beta1.Balance} returns this
 */
proto.injective.exchange.v1beta1.Balance.prototype.setDenom = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional Deposit deposits = 3;
 * @return {?proto.injective.exchange.v1beta1.Deposit}
 */
proto.injective.exchange.v1beta1.Balance.prototype.getDeposits = function() {
  return /** @type{?proto.injective.exchange.v1beta1.Deposit} */ (
    jspb.Message.getWrapperField(this, injective_exchange_v1beta1_exchange_pb.Deposit, 3));
};


/**
 * @param {?proto.injective.exchange.v1beta1.Deposit|undefined} value
 * @return {!proto.injective.exchange.v1beta1.Balance} returns this
*/
proto.injective.exchange.v1beta1.Balance.prototype.setDeposits = function(value) {
  return jspb.Message.setWrapperField(this, 3, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.injective.exchange.v1beta1.Balance} returns this
 */
proto.injective.exchange.v1beta1.Balance.prototype.clearDeposits = function() {
  return this.setDeposits(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.injective.exchange.v1beta1.Balance.prototype.hasDeposits = function() {
  return jspb.Message.getField(this, 3) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.exchange.v1beta1.DerivativePosition.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.exchange.v1beta1.DerivativePosition.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.exchange.v1beta1.DerivativePosition} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.DerivativePosition.toObject = function(includeInstance, msg) {
  var f, obj = {
    subaccountId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    marketId: jspb.Message.getFieldWithDefault(msg, 2, ""),
    position: (f = msg.getPosition()) && injective_exchange_v1beta1_exchange_pb.Position.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.exchange.v1beta1.DerivativePosition}
 */
proto.injective.exchange.v1beta1.DerivativePosition.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.exchange.v1beta1.DerivativePosition;
  return proto.injective.exchange.v1beta1.DerivativePosition.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.exchange.v1beta1.DerivativePosition} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.exchange.v1beta1.DerivativePosition}
 */
proto.injective.exchange.v1beta1.DerivativePosition.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setSubaccountId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setMarketId(value);
      break;
    case 3:
      var value = new injective_exchange_v1beta1_exchange_pb.Position;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.Position.deserializeBinaryFromReader);
      msg.setPosition(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.exchange.v1beta1.DerivativePosition.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.exchange.v1beta1.DerivativePosition.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.exchange.v1beta1.DerivativePosition} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.DerivativePosition.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getSubaccountId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getMarketId();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getPosition();
  if (f != null) {
    writer.writeMessage(
      3,
      f,
      injective_exchange_v1beta1_exchange_pb.Position.serializeBinaryToWriter
    );
  }
};


/**
 * optional string subaccount_id = 1;
 * @return {string}
 */
proto.injective.exchange.v1beta1.DerivativePosition.prototype.getSubaccountId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.exchange.v1beta1.DerivativePosition} returns this
 */
proto.injective.exchange.v1beta1.DerivativePosition.prototype.setSubaccountId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string market_id = 2;
 * @return {string}
 */
proto.injective.exchange.v1beta1.DerivativePosition.prototype.getMarketId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.exchange.v1beta1.DerivativePosition} returns this
 */
proto.injective.exchange.v1beta1.DerivativePosition.prototype.setMarketId = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional Position position = 3;
 * @return {?proto.injective.exchange.v1beta1.Position}
 */
proto.injective.exchange.v1beta1.DerivativePosition.prototype.getPosition = function() {
  return /** @type{?proto.injective.exchange.v1beta1.Position} */ (
    jspb.Message.getWrapperField(this, injective_exchange_v1beta1_exchange_pb.Position, 3));
};


/**
 * @param {?proto.injective.exchange.v1beta1.Position|undefined} value
 * @return {!proto.injective.exchange.v1beta1.DerivativePosition} returns this
*/
proto.injective.exchange.v1beta1.DerivativePosition.prototype.setPosition = function(value) {
  return jspb.Message.setWrapperField(this, 3, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.injective.exchange.v1beta1.DerivativePosition} returns this
 */
proto.injective.exchange.v1beta1.DerivativePosition.prototype.clearPosition = function() {
  return this.setPosition(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.injective.exchange.v1beta1.DerivativePosition.prototype.hasPosition = function() {
  return jspb.Message.getField(this, 3) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.exchange.v1beta1.SubaccountNonce.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.exchange.v1beta1.SubaccountNonce.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.exchange.v1beta1.SubaccountNonce} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.SubaccountNonce.toObject = function(includeInstance, msg) {
  var f, obj = {
    subaccountId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    subaccountTradeNonce: (f = msg.getSubaccountTradeNonce()) && injective_exchange_v1beta1_exchange_pb.SubaccountTradeNonce.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.exchange.v1beta1.SubaccountNonce}
 */
proto.injective.exchange.v1beta1.SubaccountNonce.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.exchange.v1beta1.SubaccountNonce;
  return proto.injective.exchange.v1beta1.SubaccountNonce.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.exchange.v1beta1.SubaccountNonce} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.exchange.v1beta1.SubaccountNonce}
 */
proto.injective.exchange.v1beta1.SubaccountNonce.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setSubaccountId(value);
      break;
    case 2:
      var value = new injective_exchange_v1beta1_exchange_pb.SubaccountTradeNonce;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.SubaccountTradeNonce.deserializeBinaryFromReader);
      msg.setSubaccountTradeNonce(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.exchange.v1beta1.SubaccountNonce.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.exchange.v1beta1.SubaccountNonce.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.exchange.v1beta1.SubaccountNonce} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.SubaccountNonce.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getSubaccountId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getSubaccountTradeNonce();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      injective_exchange_v1beta1_exchange_pb.SubaccountTradeNonce.serializeBinaryToWriter
    );
  }
};


/**
 * optional string subaccount_id = 1;
 * @return {string}
 */
proto.injective.exchange.v1beta1.SubaccountNonce.prototype.getSubaccountId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.exchange.v1beta1.SubaccountNonce} returns this
 */
proto.injective.exchange.v1beta1.SubaccountNonce.prototype.setSubaccountId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional SubaccountTradeNonce subaccount_trade_nonce = 2;
 * @return {?proto.injective.exchange.v1beta1.SubaccountTradeNonce}
 */
proto.injective.exchange.v1beta1.SubaccountNonce.prototype.getSubaccountTradeNonce = function() {
  return /** @type{?proto.injective.exchange.v1beta1.SubaccountTradeNonce} */ (
    jspb.Message.getWrapperField(this, injective_exchange_v1beta1_exchange_pb.SubaccountTradeNonce, 2));
};


/**
 * @param {?proto.injective.exchange.v1beta1.SubaccountTradeNonce|undefined} value
 * @return {!proto.injective.exchange.v1beta1.SubaccountNonce} returns this
*/
proto.injective.exchange.v1beta1.SubaccountNonce.prototype.setSubaccountTradeNonce = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.injective.exchange.v1beta1.SubaccountNonce} returns this
 */
proto.injective.exchange.v1beta1.SubaccountNonce.prototype.clearSubaccountTradeNonce = function() {
  return this.setSubaccountTradeNonce(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.injective.exchange.v1beta1.SubaccountNonce.prototype.hasSubaccountTradeNonce = function() {
  return jspb.Message.getField(this, 2) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.toObject = function(includeInstance, msg) {
  var f, obj = {
    marketId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    marketInfo: (f = msg.getMarketInfo()) && injective_exchange_v1beta1_exchange_pb.ExpiryFuturesMarketInfo.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState}
 */
proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState;
  return proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState}
 */
proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setMarketId(value);
      break;
    case 2:
      var value = new injective_exchange_v1beta1_exchange_pb.ExpiryFuturesMarketInfo;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.ExpiryFuturesMarketInfo.deserializeBinaryFromReader);
      msg.setMarketInfo(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getMarketId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getMarketInfo();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      injective_exchange_v1beta1_exchange_pb.ExpiryFuturesMarketInfo.serializeBinaryToWriter
    );
  }
};


/**
 * optional string market_id = 1;
 * @return {string}
 */
proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.prototype.getMarketId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState} returns this
 */
proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.prototype.setMarketId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional ExpiryFuturesMarketInfo market_info = 2;
 * @return {?proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfo}
 */
proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.prototype.getMarketInfo = function() {
  return /** @type{?proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfo} */ (
    jspb.Message.getWrapperField(this, injective_exchange_v1beta1_exchange_pb.ExpiryFuturesMarketInfo, 2));
};


/**
 * @param {?proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfo|undefined} value
 * @return {!proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState} returns this
*/
proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.prototype.setMarketInfo = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState} returns this
 */
proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.prototype.clearMarketInfo = function() {
  return this.setMarketInfo(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.injective.exchange.v1beta1.ExpiryFuturesMarketInfoState.prototype.hasMarketInfo = function() {
  return jspb.Message.getField(this, 2) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.exchange.v1beta1.PerpetualMarketFundingState.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.exchange.v1beta1.PerpetualMarketFundingState.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.exchange.v1beta1.PerpetualMarketFundingState} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.PerpetualMarketFundingState.toObject = function(includeInstance, msg) {
  var f, obj = {
    marketId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    funding: (f = msg.getFunding()) && injective_exchange_v1beta1_exchange_pb.PerpetualMarketFunding.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.exchange.v1beta1.PerpetualMarketFundingState}
 */
proto.injective.exchange.v1beta1.PerpetualMarketFundingState.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.exchange.v1beta1.PerpetualMarketFundingState;
  return proto.injective.exchange.v1beta1.PerpetualMarketFundingState.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.exchange.v1beta1.PerpetualMarketFundingState} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.exchange.v1beta1.PerpetualMarketFundingState}
 */
proto.injective.exchange.v1beta1.PerpetualMarketFundingState.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setMarketId(value);
      break;
    case 2:
      var value = new injective_exchange_v1beta1_exchange_pb.PerpetualMarketFunding;
      reader.readMessage(value,injective_exchange_v1beta1_exchange_pb.PerpetualMarketFunding.deserializeBinaryFromReader);
      msg.setFunding(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.exchange.v1beta1.PerpetualMarketFundingState.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.exchange.v1beta1.PerpetualMarketFundingState.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.exchange.v1beta1.PerpetualMarketFundingState} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.exchange.v1beta1.PerpetualMarketFundingState.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getMarketId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getFunding();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      injective_exchange_v1beta1_exchange_pb.PerpetualMarketFunding.serializeBinaryToWriter
    );
  }
};


/**
 * optional string market_id = 1;
 * @return {string}
 */
proto.injective.exchange.v1beta1.PerpetualMarketFundingState.prototype.getMarketId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.exchange.v1beta1.PerpetualMarketFundingState} returns this
 */
proto.injective.exchange.v1beta1.PerpetualMarketFundingState.prototype.setMarketId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional PerpetualMarketFunding funding = 2;
 * @return {?proto.injective.exchange.v1beta1.PerpetualMarketFunding}
 */
proto.injective.exchange.v1beta1.PerpetualMarketFundingState.prototype.getFunding = function() {
  return /** @type{?proto.injective.exchange.v1beta1.PerpetualMarketFunding} */ (
    jspb.Message.getWrapperField(this, injective_exchange_v1beta1_exchange_pb.PerpetualMarketFunding, 2));
};


/**
 * @param {?proto.injective.exchange.v1beta1.PerpetualMarketFunding|undefined} value
 * @return {!proto.injective.exchange.v1beta1.PerpetualMarketFundingState} returns this
*/
proto.injective.exchange.v1beta1.PerpetualMarketFundingState.prototype.setFunding = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.injective.exchange.v1beta1.PerpetualMarketFundingState} returns this
 */
proto.injective.exchange.v1beta1.PerpetualMarketFundingState.prototype.clearFunding = function() {
  return this.setFunding(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.injective.exchange.v1beta1.PerpetualMarketFundingState.prototype.hasFunding = function() {
  return jspb.Message.getField(this, 2) != null;
};


goog.object.extend(exports, proto.injective.exchange.v1beta1);
