// @injectivelabs/chain-api/injective/peggy/v1/events_pb.js
// source: injective/peggy/v1/events.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() {
  if (this) { return this; }
  if (typeof window !== 'undefined') { return window; }
  if (typeof global !== 'undefined') { return global; }
  if (typeof self !== 'undefined') { return self; }
  return Function('return this')();
}.call(null));

var gogoproto_gogo_pb = require('../../../gogoproto/gogo_pb.js');
goog.object.extend(proto, gogoproto_gogo_pb);
var injective_peggy_v1_attestation_pb = require('../../../injective/peggy/v1/attestation_pb.js');
goog.object.extend(proto, injective_peggy_v1_attestation_pb);
var injective_peggy_v1_types_pb = require('../../../injective/peggy/v1/types_pb.js');
goog.object.extend(proto, injective_peggy_v1_types_pb);
goog.exportSymbol('proto.injective.peggy.v1.EventAttestationObserved', null, global);
goog.exportSymbol('proto.injective.peggy.v1.EventAttestationVote', null, global);
goog.exportSymbol('proto.injective.peggy.v1.EventBridgeWithdrawCanceled', null, global);
goog.exportSymbol('proto.injective.peggy.v1.EventCancelSendToEth', null, global);
goog.exportSymbol('proto.injective.peggy.v1.EventConfirmBatch', null, global);
goog.exportSymbol('proto.injective.peggy.v1.EventDepositClaim', null, global);
goog.exportSymbol('proto.injective.peggy.v1.EventERC20DeployedClaim', null, global);
goog.exportSymbol('proto.injective.peggy.v1.EventOutgoingBatch', null, global);
goog.exportSymbol('proto.injective.peggy.v1.EventOutgoingBatchCanceled', null, global);
goog.exportSymbol('proto.injective.peggy.v1.EventSendToEth', null, global);
goog.exportSymbol('proto.injective.peggy.v1.EventSetOrchestratorAddresses', null, global);
goog.exportSymbol('proto.injective.peggy.v1.EventSubmitBadSignatureEvidence', null, global);
goog.exportSymbol('proto.injective.peggy.v1.EventValsetConfirm', null, global);
goog.exportSymbol('proto.injective.peggy.v1.EventValsetUpdateClaim', null, global);
goog.exportSymbol('proto.injective.peggy.v1.EventValsetUpdateRequest', null, global);
goog.exportSymbol('proto.injective.peggy.v1.EventWithdrawClaim', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.peggy.v1.EventAttestationObserved = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.peggy.v1.EventAttestationObserved, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.peggy.v1.EventAttestationObserved.displayName = 'proto.injective.peggy.v1.EventAttestationObserved';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.peggy.v1.EventBridgeWithdrawCanceled = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.peggy.v1.EventBridgeWithdrawCanceled, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.peggy.v1.EventBridgeWithdrawCanceled.displayName = 'proto.injective.peggy.v1.EventBridgeWithdrawCanceled';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.peggy.v1.EventOutgoingBatch = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.injective.peggy.v1.EventOutgoingBatch.repeatedFields_, null);
};
goog.inherits(proto.injective.peggy.v1.EventOutgoingBatch, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.peggy.v1.EventOutgoingBatch.displayName = 'proto.injective.peggy.v1.EventOutgoingBatch';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.peggy.v1.EventOutgoingBatchCanceled = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.peggy.v1.EventOutgoingBatchCanceled, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.peggy.v1.EventOutgoingBatchCanceled.displayName = 'proto.injective.peggy.v1.EventOutgoingBatchCanceled';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.peggy.v1.EventValsetUpdateRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.injective.peggy.v1.EventValsetUpdateRequest.repeatedFields_, null);
};
goog.inherits(proto.injective.peggy.v1.EventValsetUpdateRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.peggy.v1.EventValsetUpdateRequest.displayName = 'proto.injective.peggy.v1.EventValsetUpdateRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.peggy.v1.EventSetOrchestratorAddresses = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.peggy.v1.EventSetOrchestratorAddresses, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.peggy.v1.EventSetOrchestratorAddresses.displayName = 'proto.injective.peggy.v1.EventSetOrchestratorAddresses';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.peggy.v1.EventValsetConfirm = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.peggy.v1.EventValsetConfirm, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.peggy.v1.EventValsetConfirm.displayName = 'proto.injective.peggy.v1.EventValsetConfirm';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.peggy.v1.EventSendToEth = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.peggy.v1.EventSendToEth, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.peggy.v1.EventSendToEth.displayName = 'proto.injective.peggy.v1.EventSendToEth';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.peggy.v1.EventConfirmBatch = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.peggy.v1.EventConfirmBatch, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.peggy.v1.EventConfirmBatch.displayName = 'proto.injective.peggy.v1.EventConfirmBatch';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.peggy.v1.EventAttestationVote = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.peggy.v1.EventAttestationVote, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.peggy.v1.EventAttestationVote.displayName = 'proto.injective.peggy.v1.EventAttestationVote';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.peggy.v1.EventDepositClaim = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.peggy.v1.EventDepositClaim, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.peggy.v1.EventDepositClaim.displayName = 'proto.injective.peggy.v1.EventDepositClaim';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.peggy.v1.EventWithdrawClaim = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.peggy.v1.EventWithdrawClaim, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.peggy.v1.EventWithdrawClaim.displayName = 'proto.injective.peggy.v1.EventWithdrawClaim';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.peggy.v1.EventERC20DeployedClaim = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.peggy.v1.EventERC20DeployedClaim, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.peggy.v1.EventERC20DeployedClaim.displayName = 'proto.injective.peggy.v1.EventERC20DeployedClaim';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.peggy.v1.EventValsetUpdateClaim = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.injective.peggy.v1.EventValsetUpdateClaim.repeatedFields_, null);
};
goog.inherits(proto.injective.peggy.v1.EventValsetUpdateClaim, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.peggy.v1.EventValsetUpdateClaim.displayName = 'proto.injective.peggy.v1.EventValsetUpdateClaim';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.peggy.v1.EventCancelSendToEth = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.peggy.v1.EventCancelSendToEth, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.peggy.v1.EventCancelSendToEth.displayName = 'proto.injective.peggy.v1.EventCancelSendToEth';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.peggy.v1.EventSubmitBadSignatureEvidence = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.peggy.v1.EventSubmitBadSignatureEvidence, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.peggy.v1.EventSubmitBadSignatureEvidence.displayName = 'proto.injective.peggy.v1.EventSubmitBadSignatureEvidence';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.peggy.v1.EventAttestationObserved.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.peggy.v1.EventAttestationObserved.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.peggy.v1.EventAttestationObserved} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventAttestationObserved.toObject = function(includeInstance, msg) {
  var f, obj = {
    attestationType: jspb.Message.getFieldWithDefault(msg, 1, 0),
    bridgeContract: jspb.Message.getFieldWithDefault(msg, 2, ""),
    bridgeChainId: jspb.Message.getFieldWithDefault(msg, 3, 0),
    attestationId: msg.getAttestationId_asB64(),
    nonce: jspb.Message.getFieldWithDefault(msg, 5, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.peggy.v1.EventAttestationObserved}
 */
proto.injective.peggy.v1.EventAttestationObserved.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.peggy.v1.EventAttestationObserved;
  return proto.injective.peggy.v1.EventAttestationObserved.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.peggy.v1.EventAttestationObserved} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.peggy.v1.EventAttestationObserved}
 */
proto.injective.peggy.v1.EventAttestationObserved.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!proto.injective.peggy.v1.ClaimType} */ (reader.readEnum());
      msg.setAttestationType(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setBridgeContract(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setBridgeChainId(value);
      break;
    case 4:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setAttestationId(value);
      break;
    case 5:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setNonce(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventAttestationObserved.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.peggy.v1.EventAttestationObserved.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.peggy.v1.EventAttestationObserved} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventAttestationObserved.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getAttestationType();
  if (f !== 0.0) {
    writer.writeEnum(
      1,
      f
    );
  }
  f = message.getBridgeContract();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getBridgeChainId();
  if (f !== 0) {
    writer.writeUint64(
      3,
      f
    );
  }
  f = message.getAttestationId_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      4,
      f
    );
  }
  f = message.getNonce();
  if (f !== 0) {
    writer.writeUint64(
      5,
      f
    );
  }
};


/**
 * optional ClaimType attestation_type = 1;
 * @return {!proto.injective.peggy.v1.ClaimType}
 */
proto.injective.peggy.v1.EventAttestationObserved.prototype.getAttestationType = function() {
  return /** @type {!proto.injective.peggy.v1.ClaimType} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {!proto.injective.peggy.v1.ClaimType} value
 * @return {!proto.injective.peggy.v1.EventAttestationObserved} returns this
 */
proto.injective.peggy.v1.EventAttestationObserved.prototype.setAttestationType = function(value) {
  return jspb.Message.setProto3EnumField(this, 1, value);
};


/**
 * optional string bridge_contract = 2;
 * @return {string}
 */
proto.injective.peggy.v1.EventAttestationObserved.prototype.getBridgeContract = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventAttestationObserved} returns this
 */
proto.injective.peggy.v1.EventAttestationObserved.prototype.setBridgeContract = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional uint64 bridge_chain_id = 3;
 * @return {number}
 */
proto.injective.peggy.v1.EventAttestationObserved.prototype.getBridgeChainId = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventAttestationObserved} returns this
 */
proto.injective.peggy.v1.EventAttestationObserved.prototype.setBridgeChainId = function(value) {
  return jspb.Message.setProto3IntField(this, 3, value);
};


/**
 * optional bytes attestation_id = 4;
 * @return {!(string|Uint8Array)}
 */
proto.injective.peggy.v1.EventAttestationObserved.prototype.getAttestationId = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * optional bytes attestation_id = 4;
 * This is a type-conversion wrapper around `getAttestationId()`
 * @return {string}
 */
proto.injective.peggy.v1.EventAttestationObserved.prototype.getAttestationId_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getAttestationId()));
};


/**
 * optional bytes attestation_id = 4;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getAttestationId()`
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventAttestationObserved.prototype.getAttestationId_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getAttestationId()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.injective.peggy.v1.EventAttestationObserved} returns this
 */
proto.injective.peggy.v1.EventAttestationObserved.prototype.setAttestationId = function(value) {
  return jspb.Message.setProto3BytesField(this, 4, value);
};


/**
 * optional uint64 nonce = 5;
 * @return {number}
 */
proto.injective.peggy.v1.EventAttestationObserved.prototype.getNonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 5, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventAttestationObserved} returns this
 */
proto.injective.peggy.v1.EventAttestationObserved.prototype.setNonce = function(value) {
  return jspb.Message.setProto3IntField(this, 5, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.peggy.v1.EventBridgeWithdrawCanceled.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.peggy.v1.EventBridgeWithdrawCanceled.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.peggy.v1.EventBridgeWithdrawCanceled} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventBridgeWithdrawCanceled.toObject = function(includeInstance, msg) {
  var f, obj = {
    bridgeContract: jspb.Message.getFieldWithDefault(msg, 1, ""),
    bridgeChainId: jspb.Message.getFieldWithDefault(msg, 2, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.peggy.v1.EventBridgeWithdrawCanceled}
 */
proto.injective.peggy.v1.EventBridgeWithdrawCanceled.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.peggy.v1.EventBridgeWithdrawCanceled;
  return proto.injective.peggy.v1.EventBridgeWithdrawCanceled.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.peggy.v1.EventBridgeWithdrawCanceled} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.peggy.v1.EventBridgeWithdrawCanceled}
 */
proto.injective.peggy.v1.EventBridgeWithdrawCanceled.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setBridgeContract(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setBridgeChainId(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventBridgeWithdrawCanceled.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.peggy.v1.EventBridgeWithdrawCanceled.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.peggy.v1.EventBridgeWithdrawCanceled} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventBridgeWithdrawCanceled.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getBridgeContract();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getBridgeChainId();
  if (f !== 0) {
    writer.writeUint64(
      2,
      f
    );
  }
};


/**
 * optional string bridge_contract = 1;
 * @return {string}
 */
proto.injective.peggy.v1.EventBridgeWithdrawCanceled.prototype.getBridgeContract = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventBridgeWithdrawCanceled} returns this
 */
proto.injective.peggy.v1.EventBridgeWithdrawCanceled.prototype.setBridgeContract = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional uint64 bridge_chain_id = 2;
 * @return {number}
 */
proto.injective.peggy.v1.EventBridgeWithdrawCanceled.prototype.getBridgeChainId = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventBridgeWithdrawCanceled} returns this
 */
proto.injective.peggy.v1.EventBridgeWithdrawCanceled.prototype.setBridgeChainId = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.injective.peggy.v1.EventOutgoingBatch.repeatedFields_ = [5];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.peggy.v1.EventOutgoingBatch.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.peggy.v1.EventOutgoingBatch.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.peggy.v1.EventOutgoingBatch} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventOutgoingBatch.toObject = function(includeInstance, msg) {
  var f, obj = {
    denom: jspb.Message.getFieldWithDefault(msg, 1, ""),
    orchestratorAddress: jspb.Message.getFieldWithDefault(msg, 2, ""),
    batchNonce: jspb.Message.getFieldWithDefault(msg, 3, 0),
    batchTimeout: jspb.Message.getFieldWithDefault(msg, 4, 0),
    batchTxIdsList: (f = jspb.Message.getRepeatedField(msg, 5)) == null ? undefined : f
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.peggy.v1.EventOutgoingBatch}
 */
proto.injective.peggy.v1.EventOutgoingBatch.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.peggy.v1.EventOutgoingBatch;
  return proto.injective.peggy.v1.EventOutgoingBatch.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.peggy.v1.EventOutgoingBatch} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.peggy.v1.EventOutgoingBatch}
 */
proto.injective.peggy.v1.EventOutgoingBatch.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setDenom(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrchestratorAddress(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setBatchNonce(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setBatchTimeout(value);
      break;
    case 5:
      var values = /** @type {!Array<number>} */ (reader.isDelimited() ? reader.readPackedUint64() : [reader.readUint64()]);
      for (var i = 0; i < values.length; i++) {
        msg.addBatchTxIds(values[i]);
      }
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventOutgoingBatch.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.peggy.v1.EventOutgoingBatch.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.peggy.v1.EventOutgoingBatch} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventOutgoingBatch.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getDenom();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getOrchestratorAddress();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getBatchNonce();
  if (f !== 0) {
    writer.writeUint64(
      3,
      f
    );
  }
  f = message.getBatchTimeout();
  if (f !== 0) {
    writer.writeUint64(
      4,
      f
    );
  }
  f = message.getBatchTxIdsList();
  if (f.length > 0) {
    writer.writePackedUint64(
      5,
      f
    );
  }
};


/**
 * optional string denom = 1;
 * @return {string}
 */
proto.injective.peggy.v1.EventOutgoingBatch.prototype.getDenom = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventOutgoingBatch} returns this
 */
proto.injective.peggy.v1.EventOutgoingBatch.prototype.setDenom = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string orchestrator_address = 2;
 * @return {string}
 */
proto.injective.peggy.v1.EventOutgoingBatch.prototype.getOrchestratorAddress = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventOutgoingBatch} returns this
 */
proto.injective.peggy.v1.EventOutgoingBatch.prototype.setOrchestratorAddress = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional uint64 batch_nonce = 3;
 * @return {number}
 */
proto.injective.peggy.v1.EventOutgoingBatch.prototype.getBatchNonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventOutgoingBatch} returns this
 */
proto.injective.peggy.v1.EventOutgoingBatch.prototype.setBatchNonce = function(value) {
  return jspb.Message.setProto3IntField(this, 3, value);
};


/**
 * optional uint64 batch_timeout = 4;
 * @return {number}
 */
proto.injective.peggy.v1.EventOutgoingBatch.prototype.getBatchTimeout = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventOutgoingBatch} returns this
 */
proto.injective.peggy.v1.EventOutgoingBatch.prototype.setBatchTimeout = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};


/**
 * repeated uint64 batch_tx_ids = 5;
 * @return {!Array<number>}
 */
proto.injective.peggy.v1.EventOutgoingBatch.prototype.getBatchTxIdsList = function() {
  return /** @type {!Array<number>} */ (jspb.Message.getRepeatedField(this, 5));
};


/**
 * @param {!Array<number>} value
 * @return {!proto.injective.peggy.v1.EventOutgoingBatch} returns this
 */
proto.injective.peggy.v1.EventOutgoingBatch.prototype.setBatchTxIdsList = function(value) {
  return jspb.Message.setField(this, 5, value || []);
};


/**
 * @param {number} value
 * @param {number=} opt_index
 * @return {!proto.injective.peggy.v1.EventOutgoingBatch} returns this
 */
proto.injective.peggy.v1.EventOutgoingBatch.prototype.addBatchTxIds = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 5, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.peggy.v1.EventOutgoingBatch} returns this
 */
proto.injective.peggy.v1.EventOutgoingBatch.prototype.clearBatchTxIdsList = function() {
  return this.setBatchTxIdsList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.peggy.v1.EventOutgoingBatchCanceled.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.peggy.v1.EventOutgoingBatchCanceled.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.peggy.v1.EventOutgoingBatchCanceled} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventOutgoingBatchCanceled.toObject = function(includeInstance, msg) {
  var f, obj = {
    bridgeContract: jspb.Message.getFieldWithDefault(msg, 1, ""),
    bridgeChainId: jspb.Message.getFieldWithDefault(msg, 2, 0),
    batchId: jspb.Message.getFieldWithDefault(msg, 3, 0),
    nonce: jspb.Message.getFieldWithDefault(msg, 4, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.peggy.v1.EventOutgoingBatchCanceled}
 */
proto.injective.peggy.v1.EventOutgoingBatchCanceled.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.peggy.v1.EventOutgoingBatchCanceled;
  return proto.injective.peggy.v1.EventOutgoingBatchCanceled.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.peggy.v1.EventOutgoingBatchCanceled} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.peggy.v1.EventOutgoingBatchCanceled}
 */
proto.injective.peggy.v1.EventOutgoingBatchCanceled.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setBridgeContract(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setBridgeChainId(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setBatchId(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setNonce(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventOutgoingBatchCanceled.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.peggy.v1.EventOutgoingBatchCanceled.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.peggy.v1.EventOutgoingBatchCanceled} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventOutgoingBatchCanceled.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getBridgeContract();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getBridgeChainId();
  if (f !== 0) {
    writer.writeUint64(
      2,
      f
    );
  }
  f = message.getBatchId();
  if (f !== 0) {
    writer.writeUint64(
      3,
      f
    );
  }
  f = message.getNonce();
  if (f !== 0) {
    writer.writeUint64(
      4,
      f
    );
  }
};


/**
 * optional string bridge_contract = 1;
 * @return {string}
 */
proto.injective.peggy.v1.EventOutgoingBatchCanceled.prototype.getBridgeContract = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventOutgoingBatchCanceled} returns this
 */
proto.injective.peggy.v1.EventOutgoingBatchCanceled.prototype.setBridgeContract = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional uint64 bridge_chain_id = 2;
 * @return {number}
 */
proto.injective.peggy.v1.EventOutgoingBatchCanceled.prototype.getBridgeChainId = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventOutgoingBatchCanceled} returns this
 */
proto.injective.peggy.v1.EventOutgoingBatchCanceled.prototype.setBridgeChainId = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};


/**
 * optional uint64 batch_id = 3;
 * @return {number}
 */
proto.injective.peggy.v1.EventOutgoingBatchCanceled.prototype.getBatchId = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventOutgoingBatchCanceled} returns this
 */
proto.injective.peggy.v1.EventOutgoingBatchCanceled.prototype.setBatchId = function(value) {
  return jspb.Message.setProto3IntField(this, 3, value);
};


/**
 * optional uint64 nonce = 4;
 * @return {number}
 */
proto.injective.peggy.v1.EventOutgoingBatchCanceled.prototype.getNonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventOutgoingBatchCanceled} returns this
 */
proto.injective.peggy.v1.EventOutgoingBatchCanceled.prototype.setNonce = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.injective.peggy.v1.EventValsetUpdateRequest.repeatedFields_ = [3];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.peggy.v1.EventValsetUpdateRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.peggy.v1.EventValsetUpdateRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.peggy.v1.EventValsetUpdateRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventValsetUpdateRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    valsetNonce: jspb.Message.getFieldWithDefault(msg, 1, 0),
    valsetHeight: jspb.Message.getFieldWithDefault(msg, 2, 0),
    valsetMembersList: jspb.Message.toObjectList(msg.getValsetMembersList(),
    injective_peggy_v1_types_pb.BridgeValidator.toObject, includeInstance),
    rewardAmount: jspb.Message.getFieldWithDefault(msg, 4, ""),
    rewardToken: jspb.Message.getFieldWithDefault(msg, 5, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.peggy.v1.EventValsetUpdateRequest}
 */
proto.injective.peggy.v1.EventValsetUpdateRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.peggy.v1.EventValsetUpdateRequest;
  return proto.injective.peggy.v1.EventValsetUpdateRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.peggy.v1.EventValsetUpdateRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.peggy.v1.EventValsetUpdateRequest}
 */
proto.injective.peggy.v1.EventValsetUpdateRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setValsetNonce(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setValsetHeight(value);
      break;
    case 3:
      var value = new injective_peggy_v1_types_pb.BridgeValidator;
      reader.readMessage(value,injective_peggy_v1_types_pb.BridgeValidator.deserializeBinaryFromReader);
      msg.addValsetMembers(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.setRewardAmount(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setRewardToken(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventValsetUpdateRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.peggy.v1.EventValsetUpdateRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.peggy.v1.EventValsetUpdateRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventValsetUpdateRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getValsetNonce();
  if (f !== 0) {
    writer.writeUint64(
      1,
      f
    );
  }
  f = message.getValsetHeight();
  if (f !== 0) {
    writer.writeUint64(
      2,
      f
    );
  }
  f = message.getValsetMembersList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      3,
      f,
      injective_peggy_v1_types_pb.BridgeValidator.serializeBinaryToWriter
    );
  }
  f = message.getRewardAmount();
  if (f.length > 0) {
    writer.writeString(
      4,
      f
    );
  }
  f = message.getRewardToken();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
};


/**
 * optional uint64 valset_nonce = 1;
 * @return {number}
 */
proto.injective.peggy.v1.EventValsetUpdateRequest.prototype.getValsetNonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventValsetUpdateRequest} returns this
 */
proto.injective.peggy.v1.EventValsetUpdateRequest.prototype.setValsetNonce = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional uint64 valset_height = 2;
 * @return {number}
 */
proto.injective.peggy.v1.EventValsetUpdateRequest.prototype.getValsetHeight = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventValsetUpdateRequest} returns this
 */
proto.injective.peggy.v1.EventValsetUpdateRequest.prototype.setValsetHeight = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};


/**
 * repeated BridgeValidator valset_members = 3;
 * @return {!Array<!proto.injective.peggy.v1.BridgeValidator>}
 */
proto.injective.peggy.v1.EventValsetUpdateRequest.prototype.getValsetMembersList = function() {
  return /** @type{!Array<!proto.injective.peggy.v1.BridgeValidator>} */ (
    jspb.Message.getRepeatedWrapperField(this, injective_peggy_v1_types_pb.BridgeValidator, 3));
};


/**
 * @param {!Array<!proto.injective.peggy.v1.BridgeValidator>} value
 * @return {!proto.injective.peggy.v1.EventValsetUpdateRequest} returns this
*/
proto.injective.peggy.v1.EventValsetUpdateRequest.prototype.setValsetMembersList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 3, value);
};


/**
 * @param {!proto.injective.peggy.v1.BridgeValidator=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.peggy.v1.BridgeValidator}
 */
proto.injective.peggy.v1.EventValsetUpdateRequest.prototype.addValsetMembers = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.injective.peggy.v1.BridgeValidator, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.peggy.v1.EventValsetUpdateRequest} returns this
 */
proto.injective.peggy.v1.EventValsetUpdateRequest.prototype.clearValsetMembersList = function() {
  return this.setValsetMembersList([]);
};


/**
 * optional string reward_amount = 4;
 * @return {string}
 */
proto.injective.peggy.v1.EventValsetUpdateRequest.prototype.getRewardAmount = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventValsetUpdateRequest} returns this
 */
proto.injective.peggy.v1.EventValsetUpdateRequest.prototype.setRewardAmount = function(value) {
  return jspb.Message.setProto3StringField(this, 4, value);
};


/**
 * optional string reward_token = 5;
 * @return {string}
 */
proto.injective.peggy.v1.EventValsetUpdateRequest.prototype.getRewardToken = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventValsetUpdateRequest} returns this
 */
proto.injective.peggy.v1.EventValsetUpdateRequest.prototype.setRewardToken = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.peggy.v1.EventSetOrchestratorAddresses.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.peggy.v1.EventSetOrchestratorAddresses.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.peggy.v1.EventSetOrchestratorAddresses} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventSetOrchestratorAddresses.toObject = function(includeInstance, msg) {
  var f, obj = {
    validatorAddress: jspb.Message.getFieldWithDefault(msg, 1, ""),
    orchestratorAddress: jspb.Message.getFieldWithDefault(msg, 2, ""),
    operatorEthAddress: jspb.Message.getFieldWithDefault(msg, 3, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.peggy.v1.EventSetOrchestratorAddresses}
 */
proto.injective.peggy.v1.EventSetOrchestratorAddresses.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.peggy.v1.EventSetOrchestratorAddresses;
  return proto.injective.peggy.v1.EventSetOrchestratorAddresses.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.peggy.v1.EventSetOrchestratorAddresses} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.peggy.v1.EventSetOrchestratorAddresses}
 */
proto.injective.peggy.v1.EventSetOrchestratorAddresses.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setValidatorAddress(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrchestratorAddress(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setOperatorEthAddress(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventSetOrchestratorAddresses.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.peggy.v1.EventSetOrchestratorAddresses.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.peggy.v1.EventSetOrchestratorAddresses} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventSetOrchestratorAddresses.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getValidatorAddress();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getOrchestratorAddress();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getOperatorEthAddress();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
};


/**
 * optional string validator_address = 1;
 * @return {string}
 */
proto.injective.peggy.v1.EventSetOrchestratorAddresses.prototype.getValidatorAddress = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventSetOrchestratorAddresses} returns this
 */
proto.injective.peggy.v1.EventSetOrchestratorAddresses.prototype.setValidatorAddress = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string orchestrator_address = 2;
 * @return {string}
 */
proto.injective.peggy.v1.EventSetOrchestratorAddresses.prototype.getOrchestratorAddress = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventSetOrchestratorAddresses} returns this
 */
proto.injective.peggy.v1.EventSetOrchestratorAddresses.prototype.setOrchestratorAddress = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string operator_eth_address = 3;
 * @return {string}
 */
proto.injective.peggy.v1.EventSetOrchestratorAddresses.prototype.getOperatorEthAddress = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventSetOrchestratorAddresses} returns this
 */
proto.injective.peggy.v1.EventSetOrchestratorAddresses.prototype.setOperatorEthAddress = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.peggy.v1.EventValsetConfirm.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.peggy.v1.EventValsetConfirm.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.peggy.v1.EventValsetConfirm} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventValsetConfirm.toObject = function(includeInstance, msg) {
  var f, obj = {
    valsetNonce: jspb.Message.getFieldWithDefault(msg, 1, 0),
    orchestratorAddress: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.peggy.v1.EventValsetConfirm}
 */
proto.injective.peggy.v1.EventValsetConfirm.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.peggy.v1.EventValsetConfirm;
  return proto.injective.peggy.v1.EventValsetConfirm.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.peggy.v1.EventValsetConfirm} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.peggy.v1.EventValsetConfirm}
 */
proto.injective.peggy.v1.EventValsetConfirm.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setValsetNonce(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrchestratorAddress(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventValsetConfirm.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.peggy.v1.EventValsetConfirm.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.peggy.v1.EventValsetConfirm} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventValsetConfirm.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getValsetNonce();
  if (f !== 0) {
    writer.writeUint64(
      1,
      f
    );
  }
  f = message.getOrchestratorAddress();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional uint64 valset_nonce = 1;
 * @return {number}
 */
proto.injective.peggy.v1.EventValsetConfirm.prototype.getValsetNonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventValsetConfirm} returns this
 */
proto.injective.peggy.v1.EventValsetConfirm.prototype.setValsetNonce = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional string orchestrator_address = 2;
 * @return {string}
 */
proto.injective.peggy.v1.EventValsetConfirm.prototype.getOrchestratorAddress = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventValsetConfirm} returns this
 */
proto.injective.peggy.v1.EventValsetConfirm.prototype.setOrchestratorAddress = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.peggy.v1.EventSendToEth.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.peggy.v1.EventSendToEth.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.peggy.v1.EventSendToEth} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventSendToEth.toObject = function(includeInstance, msg) {
  var f, obj = {
    outgoingTxId: jspb.Message.getFieldWithDefault(msg, 1, 0),
    sender: jspb.Message.getFieldWithDefault(msg, 2, ""),
    receiver: jspb.Message.getFieldWithDefault(msg, 3, ""),
    amount: jspb.Message.getFieldWithDefault(msg, 4, ""),
    bridgeFee: jspb.Message.getFieldWithDefault(msg, 5, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.peggy.v1.EventSendToEth}
 */
proto.injective.peggy.v1.EventSendToEth.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.peggy.v1.EventSendToEth;
  return proto.injective.peggy.v1.EventSendToEth.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.peggy.v1.EventSendToEth} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.peggy.v1.EventSendToEth}
 */
proto.injective.peggy.v1.EventSendToEth.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setOutgoingTxId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setSender(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setReceiver(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.setAmount(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setBridgeFee(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventSendToEth.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.peggy.v1.EventSendToEth.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.peggy.v1.EventSendToEth} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventSendToEth.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getOutgoingTxId();
  if (f !== 0) {
    writer.writeUint64(
      1,
      f
    );
  }
  f = message.getSender();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getReceiver();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getAmount();
  if (f.length > 0) {
    writer.writeString(
      4,
      f
    );
  }
  f = message.getBridgeFee();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
};


/**
 * optional uint64 outgoing_tx_id = 1;
 * @return {number}
 */
proto.injective.peggy.v1.EventSendToEth.prototype.getOutgoingTxId = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventSendToEth} returns this
 */
proto.injective.peggy.v1.EventSendToEth.prototype.setOutgoingTxId = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional string sender = 2;
 * @return {string}
 */
proto.injective.peggy.v1.EventSendToEth.prototype.getSender = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventSendToEth} returns this
 */
proto.injective.peggy.v1.EventSendToEth.prototype.setSender = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string receiver = 3;
 * @return {string}
 */
proto.injective.peggy.v1.EventSendToEth.prototype.getReceiver = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventSendToEth} returns this
 */
proto.injective.peggy.v1.EventSendToEth.prototype.setReceiver = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional string amount = 4;
 * @return {string}
 */
proto.injective.peggy.v1.EventSendToEth.prototype.getAmount = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventSendToEth} returns this
 */
proto.injective.peggy.v1.EventSendToEth.prototype.setAmount = function(value) {
  return jspb.Message.setProto3StringField(this, 4, value);
};


/**
 * optional string bridge_fee = 5;
 * @return {string}
 */
proto.injective.peggy.v1.EventSendToEth.prototype.getBridgeFee = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventSendToEth} returns this
 */
proto.injective.peggy.v1.EventSendToEth.prototype.setBridgeFee = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.peggy.v1.EventConfirmBatch.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.peggy.v1.EventConfirmBatch.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.peggy.v1.EventConfirmBatch} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventConfirmBatch.toObject = function(includeInstance, msg) {
  var f, obj = {
    batchNonce: jspb.Message.getFieldWithDefault(msg, 1, 0),
    orchestratorAddress: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.peggy.v1.EventConfirmBatch}
 */
proto.injective.peggy.v1.EventConfirmBatch.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.peggy.v1.EventConfirmBatch;
  return proto.injective.peggy.v1.EventConfirmBatch.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.peggy.v1.EventConfirmBatch} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.peggy.v1.EventConfirmBatch}
 */
proto.injective.peggy.v1.EventConfirmBatch.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setBatchNonce(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrchestratorAddress(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventConfirmBatch.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.peggy.v1.EventConfirmBatch.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.peggy.v1.EventConfirmBatch} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventConfirmBatch.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getBatchNonce();
  if (f !== 0) {
    writer.writeUint64(
      1,
      f
    );
  }
  f = message.getOrchestratorAddress();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional uint64 batch_nonce = 1;
 * @return {number}
 */
proto.injective.peggy.v1.EventConfirmBatch.prototype.getBatchNonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventConfirmBatch} returns this
 */
proto.injective.peggy.v1.EventConfirmBatch.prototype.setBatchNonce = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional string orchestrator_address = 2;
 * @return {string}
 */
proto.injective.peggy.v1.EventConfirmBatch.prototype.getOrchestratorAddress = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventConfirmBatch} returns this
 */
proto.injective.peggy.v1.EventConfirmBatch.prototype.setOrchestratorAddress = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.peggy.v1.EventAttestationVote.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.peggy.v1.EventAttestationVote.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.peggy.v1.EventAttestationVote} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventAttestationVote.toObject = function(includeInstance, msg) {
  var f, obj = {
    eventNonce: jspb.Message.getFieldWithDefault(msg, 1, 0),
    attestationId: msg.getAttestationId_asB64(),
    voter: jspb.Message.getFieldWithDefault(msg, 3, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.peggy.v1.EventAttestationVote}
 */
proto.injective.peggy.v1.EventAttestationVote.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.peggy.v1.EventAttestationVote;
  return proto.injective.peggy.v1.EventAttestationVote.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.peggy.v1.EventAttestationVote} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.peggy.v1.EventAttestationVote}
 */
proto.injective.peggy.v1.EventAttestationVote.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setEventNonce(value);
      break;
    case 2:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setAttestationId(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setVoter(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventAttestationVote.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.peggy.v1.EventAttestationVote.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.peggy.v1.EventAttestationVote} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventAttestationVote.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getEventNonce();
  if (f !== 0) {
    writer.writeUint64(
      1,
      f
    );
  }
  f = message.getAttestationId_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      2,
      f
    );
  }
  f = message.getVoter();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
};


/**
 * optional uint64 event_nonce = 1;
 * @return {number}
 */
proto.injective.peggy.v1.EventAttestationVote.prototype.getEventNonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventAttestationVote} returns this
 */
proto.injective.peggy.v1.EventAttestationVote.prototype.setEventNonce = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional bytes attestation_id = 2;
 * @return {!(string|Uint8Array)}
 */
proto.injective.peggy.v1.EventAttestationVote.prototype.getAttestationId = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * optional bytes attestation_id = 2;
 * This is a type-conversion wrapper around `getAttestationId()`
 * @return {string}
 */
proto.injective.peggy.v1.EventAttestationVote.prototype.getAttestationId_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getAttestationId()));
};


/**
 * optional bytes attestation_id = 2;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getAttestationId()`
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventAttestationVote.prototype.getAttestationId_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getAttestationId()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.injective.peggy.v1.EventAttestationVote} returns this
 */
proto.injective.peggy.v1.EventAttestationVote.prototype.setAttestationId = function(value) {
  return jspb.Message.setProto3BytesField(this, 2, value);
};


/**
 * optional string voter = 3;
 * @return {string}
 */
proto.injective.peggy.v1.EventAttestationVote.prototype.getVoter = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventAttestationVote} returns this
 */
proto.injective.peggy.v1.EventAttestationVote.prototype.setVoter = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.peggy.v1.EventDepositClaim.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.peggy.v1.EventDepositClaim} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventDepositClaim.toObject = function(includeInstance, msg) {
  var f, obj = {
    eventNonce: jspb.Message.getFieldWithDefault(msg, 1, 0),
    eventHeight: jspb.Message.getFieldWithDefault(msg, 2, 0),
    attestationId: msg.getAttestationId_asB64(),
    ethereumSender: jspb.Message.getFieldWithDefault(msg, 4, ""),
    cosmosReceiver: jspb.Message.getFieldWithDefault(msg, 5, ""),
    tokenContract: jspb.Message.getFieldWithDefault(msg, 6, ""),
    amount: jspb.Message.getFieldWithDefault(msg, 7, ""),
    orchestratorAddress: jspb.Message.getFieldWithDefault(msg, 8, ""),
    data: jspb.Message.getFieldWithDefault(msg, 9, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.peggy.v1.EventDepositClaim}
 */
proto.injective.peggy.v1.EventDepositClaim.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.peggy.v1.EventDepositClaim;
  return proto.injective.peggy.v1.EventDepositClaim.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.peggy.v1.EventDepositClaim} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.peggy.v1.EventDepositClaim}
 */
proto.injective.peggy.v1.EventDepositClaim.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setEventNonce(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setEventHeight(value);
      break;
    case 3:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setAttestationId(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.setEthereumSender(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setCosmosReceiver(value);
      break;
    case 6:
      var value = /** @type {string} */ (reader.readString());
      msg.setTokenContract(value);
      break;
    case 7:
      var value = /** @type {string} */ (reader.readString());
      msg.setAmount(value);
      break;
    case 8:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrchestratorAddress(value);
      break;
    case 9:
      var value = /** @type {string} */ (reader.readString());
      msg.setData(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.peggy.v1.EventDepositClaim.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.peggy.v1.EventDepositClaim} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventDepositClaim.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getEventNonce();
  if (f !== 0) {
    writer.writeUint64(
      1,
      f
    );
  }
  f = message.getEventHeight();
  if (f !== 0) {
    writer.writeUint64(
      2,
      f
    );
  }
  f = message.getAttestationId_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      3,
      f
    );
  }
  f = message.getEthereumSender();
  if (f.length > 0) {
    writer.writeString(
      4,
      f
    );
  }
  f = message.getCosmosReceiver();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getTokenContract();
  if (f.length > 0) {
    writer.writeString(
      6,
      f
    );
  }
  f = message.getAmount();
  if (f.length > 0) {
    writer.writeString(
      7,
      f
    );
  }
  f = message.getOrchestratorAddress();
  if (f.length > 0) {
    writer.writeString(
      8,
      f
    );
  }
  f = message.getData();
  if (f.length > 0) {
    writer.writeString(
      9,
      f
    );
  }
};


/**
 * optional uint64 event_nonce = 1;
 * @return {number}
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.getEventNonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventDepositClaim} returns this
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.setEventNonce = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional uint64 event_height = 2;
 * @return {number}
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.getEventHeight = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventDepositClaim} returns this
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.setEventHeight = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};


/**
 * optional bytes attestation_id = 3;
 * @return {!(string|Uint8Array)}
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.getAttestationId = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * optional bytes attestation_id = 3;
 * This is a type-conversion wrapper around `getAttestationId()`
 * @return {string}
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.getAttestationId_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getAttestationId()));
};


/**
 * optional bytes attestation_id = 3;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getAttestationId()`
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.getAttestationId_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getAttestationId()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.injective.peggy.v1.EventDepositClaim} returns this
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.setAttestationId = function(value) {
  return jspb.Message.setProto3BytesField(this, 3, value);
};


/**
 * optional string ethereum_sender = 4;
 * @return {string}
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.getEthereumSender = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventDepositClaim} returns this
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.setEthereumSender = function(value) {
  return jspb.Message.setProto3StringField(this, 4, value);
};


/**
 * optional string cosmos_receiver = 5;
 * @return {string}
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.getCosmosReceiver = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventDepositClaim} returns this
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.setCosmosReceiver = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional string token_contract = 6;
 * @return {string}
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.getTokenContract = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 6, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventDepositClaim} returns this
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.setTokenContract = function(value) {
  return jspb.Message.setProto3StringField(this, 6, value);
};


/**
 * optional string amount = 7;
 * @return {string}
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.getAmount = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 7, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventDepositClaim} returns this
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.setAmount = function(value) {
  return jspb.Message.setProto3StringField(this, 7, value);
};


/**
 * optional string orchestrator_address = 8;
 * @return {string}
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.getOrchestratorAddress = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 8, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventDepositClaim} returns this
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.setOrchestratorAddress = function(value) {
  return jspb.Message.setProto3StringField(this, 8, value);
};


/**
 * optional string data = 9;
 * @return {string}
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.getData = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 9, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventDepositClaim} returns this
 */
proto.injective.peggy.v1.EventDepositClaim.prototype.setData = function(value) {
  return jspb.Message.setProto3StringField(this, 9, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.peggy.v1.EventWithdrawClaim.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.peggy.v1.EventWithdrawClaim.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.peggy.v1.EventWithdrawClaim} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventWithdrawClaim.toObject = function(includeInstance, msg) {
  var f, obj = {
    eventNonce: jspb.Message.getFieldWithDefault(msg, 1, 0),
    eventHeight: jspb.Message.getFieldWithDefault(msg, 2, 0),
    attestationId: msg.getAttestationId_asB64(),
    batchNonce: jspb.Message.getFieldWithDefault(msg, 4, 0),
    tokenContract: jspb.Message.getFieldWithDefault(msg, 5, ""),
    orchestratorAddress: jspb.Message.getFieldWithDefault(msg, 6, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.peggy.v1.EventWithdrawClaim}
 */
proto.injective.peggy.v1.EventWithdrawClaim.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.peggy.v1.EventWithdrawClaim;
  return proto.injective.peggy.v1.EventWithdrawClaim.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.peggy.v1.EventWithdrawClaim} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.peggy.v1.EventWithdrawClaim}
 */
proto.injective.peggy.v1.EventWithdrawClaim.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setEventNonce(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setEventHeight(value);
      break;
    case 3:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setAttestationId(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setBatchNonce(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setTokenContract(value);
      break;
    case 6:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrchestratorAddress(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventWithdrawClaim.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.peggy.v1.EventWithdrawClaim.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.peggy.v1.EventWithdrawClaim} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventWithdrawClaim.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getEventNonce();
  if (f !== 0) {
    writer.writeUint64(
      1,
      f
    );
  }
  f = message.getEventHeight();
  if (f !== 0) {
    writer.writeUint64(
      2,
      f
    );
  }
  f = message.getAttestationId_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      3,
      f
    );
  }
  f = message.getBatchNonce();
  if (f !== 0) {
    writer.writeUint64(
      4,
      f
    );
  }
  f = message.getTokenContract();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getOrchestratorAddress();
  if (f.length > 0) {
    writer.writeString(
      6,
      f
    );
  }
};


/**
 * optional uint64 event_nonce = 1;
 * @return {number}
 */
proto.injective.peggy.v1.EventWithdrawClaim.prototype.getEventNonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventWithdrawClaim} returns this
 */
proto.injective.peggy.v1.EventWithdrawClaim.prototype.setEventNonce = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional uint64 event_height = 2;
 * @return {number}
 */
proto.injective.peggy.v1.EventWithdrawClaim.prototype.getEventHeight = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventWithdrawClaim} returns this
 */
proto.injective.peggy.v1.EventWithdrawClaim.prototype.setEventHeight = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};


/**
 * optional bytes attestation_id = 3;
 * @return {!(string|Uint8Array)}
 */
proto.injective.peggy.v1.EventWithdrawClaim.prototype.getAttestationId = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * optional bytes attestation_id = 3;
 * This is a type-conversion wrapper around `getAttestationId()`
 * @return {string}
 */
proto.injective.peggy.v1.EventWithdrawClaim.prototype.getAttestationId_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getAttestationId()));
};


/**
 * optional bytes attestation_id = 3;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getAttestationId()`
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventWithdrawClaim.prototype.getAttestationId_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getAttestationId()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.injective.peggy.v1.EventWithdrawClaim} returns this
 */
proto.injective.peggy.v1.EventWithdrawClaim.prototype.setAttestationId = function(value) {
  return jspb.Message.setProto3BytesField(this, 3, value);
};


/**
 * optional uint64 batch_nonce = 4;
 * @return {number}
 */
proto.injective.peggy.v1.EventWithdrawClaim.prototype.getBatchNonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventWithdrawClaim} returns this
 */
proto.injective.peggy.v1.EventWithdrawClaim.prototype.setBatchNonce = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};


/**
 * optional string token_contract = 5;
 * @return {string}
 */
proto.injective.peggy.v1.EventWithdrawClaim.prototype.getTokenContract = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventWithdrawClaim} returns this
 */
proto.injective.peggy.v1.EventWithdrawClaim.prototype.setTokenContract = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional string orchestrator_address = 6;
 * @return {string}
 */
proto.injective.peggy.v1.EventWithdrawClaim.prototype.getOrchestratorAddress = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 6, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventWithdrawClaim} returns this
 */
proto.injective.peggy.v1.EventWithdrawClaim.prototype.setOrchestratorAddress = function(value) {
  return jspb.Message.setProto3StringField(this, 6, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.peggy.v1.EventERC20DeployedClaim.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.peggy.v1.EventERC20DeployedClaim} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.toObject = function(includeInstance, msg) {
  var f, obj = {
    eventNonce: jspb.Message.getFieldWithDefault(msg, 1, 0),
    eventHeight: jspb.Message.getFieldWithDefault(msg, 2, 0),
    attestationId: msg.getAttestationId_asB64(),
    cosmosDenom: jspb.Message.getFieldWithDefault(msg, 4, ""),
    tokenContract: jspb.Message.getFieldWithDefault(msg, 5, ""),
    name: jspb.Message.getFieldWithDefault(msg, 6, ""),
    symbol: jspb.Message.getFieldWithDefault(msg, 7, ""),
    decimals: jspb.Message.getFieldWithDefault(msg, 8, 0),
    orchestratorAddress: jspb.Message.getFieldWithDefault(msg, 9, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.peggy.v1.EventERC20DeployedClaim}
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.peggy.v1.EventERC20DeployedClaim;
  return proto.injective.peggy.v1.EventERC20DeployedClaim.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.peggy.v1.EventERC20DeployedClaim} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.peggy.v1.EventERC20DeployedClaim}
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setEventNonce(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setEventHeight(value);
      break;
    case 3:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setAttestationId(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.setCosmosDenom(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setTokenContract(value);
      break;
    case 6:
      var value = /** @type {string} */ (reader.readString());
      msg.setName(value);
      break;
    case 7:
      var value = /** @type {string} */ (reader.readString());
      msg.setSymbol(value);
      break;
    case 8:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setDecimals(value);
      break;
    case 9:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrchestratorAddress(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.peggy.v1.EventERC20DeployedClaim.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.peggy.v1.EventERC20DeployedClaim} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getEventNonce();
  if (f !== 0) {
    writer.writeUint64(
      1,
      f
    );
  }
  f = message.getEventHeight();
  if (f !== 0) {
    writer.writeUint64(
      2,
      f
    );
  }
  f = message.getAttestationId_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      3,
      f
    );
  }
  f = message.getCosmosDenom();
  if (f.length > 0) {
    writer.writeString(
      4,
      f
    );
  }
  f = message.getTokenContract();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getName();
  if (f.length > 0) {
    writer.writeString(
      6,
      f
    );
  }
  f = message.getSymbol();
  if (f.length > 0) {
    writer.writeString(
      7,
      f
    );
  }
  f = message.getDecimals();
  if (f !== 0) {
    writer.writeUint64(
      8,
      f
    );
  }
  f = message.getOrchestratorAddress();
  if (f.length > 0) {
    writer.writeString(
      9,
      f
    );
  }
};


/**
 * optional uint64 event_nonce = 1;
 * @return {number}
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.getEventNonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventERC20DeployedClaim} returns this
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.setEventNonce = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional uint64 event_height = 2;
 * @return {number}
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.getEventHeight = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventERC20DeployedClaim} returns this
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.setEventHeight = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};


/**
 * optional bytes attestation_id = 3;
 * @return {!(string|Uint8Array)}
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.getAttestationId = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * optional bytes attestation_id = 3;
 * This is a type-conversion wrapper around `getAttestationId()`
 * @return {string}
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.getAttestationId_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getAttestationId()));
};


/**
 * optional bytes attestation_id = 3;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getAttestationId()`
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.getAttestationId_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getAttestationId()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.injective.peggy.v1.EventERC20DeployedClaim} returns this
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.setAttestationId = function(value) {
  return jspb.Message.setProto3BytesField(this, 3, value);
};


/**
 * optional string cosmos_denom = 4;
 * @return {string}
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.getCosmosDenom = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventERC20DeployedClaim} returns this
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.setCosmosDenom = function(value) {
  return jspb.Message.setProto3StringField(this, 4, value);
};


/**
 * optional string token_contract = 5;
 * @return {string}
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.getTokenContract = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventERC20DeployedClaim} returns this
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.setTokenContract = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional string name = 6;
 * @return {string}
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.getName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 6, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventERC20DeployedClaim} returns this
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.setName = function(value) {
  return jspb.Message.setProto3StringField(this, 6, value);
};


/**
 * optional string symbol = 7;
 * @return {string}
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.getSymbol = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 7, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventERC20DeployedClaim} returns this
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.setSymbol = function(value) {
  return jspb.Message.setProto3StringField(this, 7, value);
};


/**
 * optional uint64 decimals = 8;
 * @return {number}
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.getDecimals = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 8, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventERC20DeployedClaim} returns this
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.setDecimals = function(value) {
  return jspb.Message.setProto3IntField(this, 8, value);
};


/**
 * optional string orchestrator_address = 9;
 * @return {string}
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.getOrchestratorAddress = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 9, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventERC20DeployedClaim} returns this
 */
proto.injective.peggy.v1.EventERC20DeployedClaim.prototype.setOrchestratorAddress = function(value) {
  return jspb.Message.setProto3StringField(this, 9, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.repeatedFields_ = [5];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.peggy.v1.EventValsetUpdateClaim.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.peggy.v1.EventValsetUpdateClaim} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.toObject = function(includeInstance, msg) {
  var f, obj = {
    eventNonce: jspb.Message.getFieldWithDefault(msg, 1, 0),
    eventHeight: jspb.Message.getFieldWithDefault(msg, 2, 0),
    attestationId: msg.getAttestationId_asB64(),
    valsetNonce: jspb.Message.getFieldWithDefault(msg, 4, 0),
    valsetMembersList: jspb.Message.toObjectList(msg.getValsetMembersList(),
    injective_peggy_v1_types_pb.BridgeValidator.toObject, includeInstance),
    rewardAmount: jspb.Message.getFieldWithDefault(msg, 6, ""),
    rewardToken: jspb.Message.getFieldWithDefault(msg, 7, ""),
    orchestratorAddress: jspb.Message.getFieldWithDefault(msg, 8, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.peggy.v1.EventValsetUpdateClaim}
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.peggy.v1.EventValsetUpdateClaim;
  return proto.injective.peggy.v1.EventValsetUpdateClaim.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.peggy.v1.EventValsetUpdateClaim} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.peggy.v1.EventValsetUpdateClaim}
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setEventNonce(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setEventHeight(value);
      break;
    case 3:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setAttestationId(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setValsetNonce(value);
      break;
    case 5:
      var value = new injective_peggy_v1_types_pb.BridgeValidator;
      reader.readMessage(value,injective_peggy_v1_types_pb.BridgeValidator.deserializeBinaryFromReader);
      msg.addValsetMembers(value);
      break;
    case 6:
      var value = /** @type {string} */ (reader.readString());
      msg.setRewardAmount(value);
      break;
    case 7:
      var value = /** @type {string} */ (reader.readString());
      msg.setRewardToken(value);
      break;
    case 8:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrchestratorAddress(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.peggy.v1.EventValsetUpdateClaim.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.peggy.v1.EventValsetUpdateClaim} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getEventNonce();
  if (f !== 0) {
    writer.writeUint64(
      1,
      f
    );
  }
  f = message.getEventHeight();
  if (f !== 0) {
    writer.writeUint64(
      2,
      f
    );
  }
  f = message.getAttestationId_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      3,
      f
    );
  }
  f = message.getValsetNonce();
  if (f !== 0) {
    writer.writeUint64(
      4,
      f
    );
  }
  f = message.getValsetMembersList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      5,
      f,
      injective_peggy_v1_types_pb.BridgeValidator.serializeBinaryToWriter
    );
  }
  f = message.getRewardAmount();
  if (f.length > 0) {
    writer.writeString(
      6,
      f
    );
  }
  f = message.getRewardToken();
  if (f.length > 0) {
    writer.writeString(
      7,
      f
    );
  }
  f = message.getOrchestratorAddress();
  if (f.length > 0) {
    writer.writeString(
      8,
      f
    );
  }
};


/**
 * optional uint64 event_nonce = 1;
 * @return {number}
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.getEventNonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventValsetUpdateClaim} returns this
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.setEventNonce = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional uint64 event_height = 2;
 * @return {number}
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.getEventHeight = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventValsetUpdateClaim} returns this
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.setEventHeight = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};


/**
 * optional bytes attestation_id = 3;
 * @return {!(string|Uint8Array)}
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.getAttestationId = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * optional bytes attestation_id = 3;
 * This is a type-conversion wrapper around `getAttestationId()`
 * @return {string}
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.getAttestationId_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getAttestationId()));
};


/**
 * optional bytes attestation_id = 3;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getAttestationId()`
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.getAttestationId_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getAttestationId()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.injective.peggy.v1.EventValsetUpdateClaim} returns this
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.setAttestationId = function(value) {
  return jspb.Message.setProto3BytesField(this, 3, value);
};


/**
 * optional uint64 valset_nonce = 4;
 * @return {number}
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.getValsetNonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventValsetUpdateClaim} returns this
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.setValsetNonce = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};


/**
 * repeated BridgeValidator valset_members = 5;
 * @return {!Array<!proto.injective.peggy.v1.BridgeValidator>}
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.getValsetMembersList = function() {
  return /** @type{!Array<!proto.injective.peggy.v1.BridgeValidator>} */ (
    jspb.Message.getRepeatedWrapperField(this, injective_peggy_v1_types_pb.BridgeValidator, 5));
};


/**
 * @param {!Array<!proto.injective.peggy.v1.BridgeValidator>} value
 * @return {!proto.injective.peggy.v1.EventValsetUpdateClaim} returns this
*/
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.setValsetMembersList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 5, value);
};


/**
 * @param {!proto.injective.peggy.v1.BridgeValidator=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.peggy.v1.BridgeValidator}
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.addValsetMembers = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 5, opt_value, proto.injective.peggy.v1.BridgeValidator, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.peggy.v1.EventValsetUpdateClaim} returns this
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.clearValsetMembersList = function() {
  return this.setValsetMembersList([]);
};


/**
 * optional string reward_amount = 6;
 * @return {string}
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.getRewardAmount = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 6, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventValsetUpdateClaim} returns this
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.setRewardAmount = function(value) {
  return jspb.Message.setProto3StringField(this, 6, value);
};


/**
 * optional string reward_token = 7;
 * @return {string}
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.getRewardToken = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 7, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventValsetUpdateClaim} returns this
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.setRewardToken = function(value) {
  return jspb.Message.setProto3StringField(this, 7, value);
};


/**
 * optional string orchestrator_address = 8;
 * @return {string}
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.getOrchestratorAddress = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 8, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventValsetUpdateClaim} returns this
 */
proto.injective.peggy.v1.EventValsetUpdateClaim.prototype.setOrchestratorAddress = function(value) {
  return jspb.Message.setProto3StringField(this, 8, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.peggy.v1.EventCancelSendToEth.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.peggy.v1.EventCancelSendToEth.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.peggy.v1.EventCancelSendToEth} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventCancelSendToEth.toObject = function(includeInstance, msg) {
  var f, obj = {
    outgoingTxId: jspb.Message.getFieldWithDefault(msg, 1, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.peggy.v1.EventCancelSendToEth}
 */
proto.injective.peggy.v1.EventCancelSendToEth.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.peggy.v1.EventCancelSendToEth;
  return proto.injective.peggy.v1.EventCancelSendToEth.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.peggy.v1.EventCancelSendToEth} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.peggy.v1.EventCancelSendToEth}
 */
proto.injective.peggy.v1.EventCancelSendToEth.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setOutgoingTxId(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventCancelSendToEth.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.peggy.v1.EventCancelSendToEth.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.peggy.v1.EventCancelSendToEth} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventCancelSendToEth.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getOutgoingTxId();
  if (f !== 0) {
    writer.writeUint64(
      1,
      f
    );
  }
};


/**
 * optional uint64 outgoing_tx_id = 1;
 * @return {number}
 */
proto.injective.peggy.v1.EventCancelSendToEth.prototype.getOutgoingTxId = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.injective.peggy.v1.EventCancelSendToEth} returns this
 */
proto.injective.peggy.v1.EventCancelSendToEth.prototype.setOutgoingTxId = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.peggy.v1.EventSubmitBadSignatureEvidence.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.peggy.v1.EventSubmitBadSignatureEvidence.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.peggy.v1.EventSubmitBadSignatureEvidence} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventSubmitBadSignatureEvidence.toObject = function(includeInstance, msg) {
  var f, obj = {
    badEthSignature: jspb.Message.getFieldWithDefault(msg, 1, ""),
    badEthSignatureSubject: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.peggy.v1.EventSubmitBadSignatureEvidence}
 */
proto.injective.peggy.v1.EventSubmitBadSignatureEvidence.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.peggy.v1.EventSubmitBadSignatureEvidence;
  return proto.injective.peggy.v1.EventSubmitBadSignatureEvidence.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.peggy.v1.EventSubmitBadSignatureEvidence} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.peggy.v1.EventSubmitBadSignatureEvidence}
 */
proto.injective.peggy.v1.EventSubmitBadSignatureEvidence.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setBadEthSignature(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setBadEthSignatureSubject(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.peggy.v1.EventSubmitBadSignatureEvidence.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.peggy.v1.EventSubmitBadSignatureEvidence.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.peggy.v1.EventSubmitBadSignatureEvidence} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.peggy.v1.EventSubmitBadSignatureEvidence.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getBadEthSignature();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getBadEthSignatureSubject();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional string bad_eth_signature = 1;
 * @return {string}
 */
proto.injective.peggy.v1.EventSubmitBadSignatureEvidence.prototype.getBadEthSignature = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventSubmitBadSignatureEvidence} returns this
 */
proto.injective.peggy.v1.EventSubmitBadSignatureEvidence.prototype.setBadEthSignature = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string bad_eth_signature_subject = 2;
 * @return {string}
 */
proto.injective.peggy.v1.EventSubmitBadSignatureEvidence.prototype.getBadEthSignatureSubject = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.peggy.v1.EventSubmitBadSignatureEvidence} returns this
 */
proto.injective.peggy.v1.EventSubmitBadSignatureEvidence.prototype.setBadEthSignatureSubject = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


goog.object.extend(exports, proto.injective.peggy.v1);
