// @injectivelabs/chain-api/injective/tokenfactory/v1beta1/genesis_pb.js
// source: injective/tokenfactory/v1beta1/genesis.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() {
  if (this) { return this; }
  if (typeof window !== 'undefined') { return window; }
  if (typeof global !== 'undefined') { return global; }
  if (typeof self !== 'undefined') { return self; }
  return Function('return this')();
}.call(null));

var gogoproto_gogo_pb = require('../../../gogoproto/gogo_pb.js');
goog.object.extend(proto, gogoproto_gogo_pb);
var injective_tokenfactory_v1beta1_authorityMetadata_pb = require('../../../injective/tokenfactory/v1beta1/authorityMetadata_pb.js');
goog.object.extend(proto, injective_tokenfactory_v1beta1_authorityMetadata_pb);
var injective_tokenfactory_v1beta1_params_pb = require('../../../injective/tokenfactory/v1beta1/params_pb.js');
goog.object.extend(proto, injective_tokenfactory_v1beta1_params_pb);
goog.exportSymbol('proto.injective.tokenfactory.v1beta1.GenesisDenom', null, global);
goog.exportSymbol('proto.injective.tokenfactory.v1beta1.GenesisState', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.tokenfactory.v1beta1.GenesisState = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.injective.tokenfactory.v1beta1.GenesisState.repeatedFields_, null);
};
goog.inherits(proto.injective.tokenfactory.v1beta1.GenesisState, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.tokenfactory.v1beta1.GenesisState.displayName = 'proto.injective.tokenfactory.v1beta1.GenesisState';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.tokenfactory.v1beta1.GenesisDenom = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.tokenfactory.v1beta1.GenesisDenom, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.tokenfactory.v1beta1.GenesisDenom.displayName = 'proto.injective.tokenfactory.v1beta1.GenesisDenom';
}

/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.injective.tokenfactory.v1beta1.GenesisState.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.tokenfactory.v1beta1.GenesisState.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.tokenfactory.v1beta1.GenesisState.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.tokenfactory.v1beta1.GenesisState} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.tokenfactory.v1beta1.GenesisState.toObject = function(includeInstance, msg) {
  var f, obj = {
    params: (f = msg.getParams()) && injective_tokenfactory_v1beta1_params_pb.Params.toObject(includeInstance, f),
    factoryDenomsList: jspb.Message.toObjectList(msg.getFactoryDenomsList(),
    proto.injective.tokenfactory.v1beta1.GenesisDenom.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.tokenfactory.v1beta1.GenesisState}
 */
proto.injective.tokenfactory.v1beta1.GenesisState.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.tokenfactory.v1beta1.GenesisState;
  return proto.injective.tokenfactory.v1beta1.GenesisState.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.tokenfactory.v1beta1.GenesisState} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.tokenfactory.v1beta1.GenesisState}
 */
proto.injective.tokenfactory.v1beta1.GenesisState.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new injective_tokenfactory_v1beta1_params_pb.Params;
      reader.readMessage(value,injective_tokenfactory_v1beta1_params_pb.Params.deserializeBinaryFromReader);
      msg.setParams(value);
      break;
    case 2:
      var value = new proto.injective.tokenfactory.v1beta1.GenesisDenom;
      reader.readMessage(value,proto.injective.tokenfactory.v1beta1.GenesisDenom.deserializeBinaryFromReader);
      msg.addFactoryDenoms(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.tokenfactory.v1beta1.GenesisState.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.tokenfactory.v1beta1.GenesisState.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.tokenfactory.v1beta1.GenesisState} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.tokenfactory.v1beta1.GenesisState.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getParams();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      injective_tokenfactory_v1beta1_params_pb.Params.serializeBinaryToWriter
    );
  }
  f = message.getFactoryDenomsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      proto.injective.tokenfactory.v1beta1.GenesisDenom.serializeBinaryToWriter
    );
  }
};


/**
 * optional Params params = 1;
 * @return {?proto.injective.tokenfactory.v1beta1.Params}
 */
proto.injective.tokenfactory.v1beta1.GenesisState.prototype.getParams = function() {
  return /** @type{?proto.injective.tokenfactory.v1beta1.Params} */ (
    jspb.Message.getWrapperField(this, injective_tokenfactory_v1beta1_params_pb.Params, 1));
};


/**
 * @param {?proto.injective.tokenfactory.v1beta1.Params|undefined} value
 * @return {!proto.injective.tokenfactory.v1beta1.GenesisState} returns this
*/
proto.injective.tokenfactory.v1beta1.GenesisState.prototype.setParams = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.injective.tokenfactory.v1beta1.GenesisState} returns this
 */
proto.injective.tokenfactory.v1beta1.GenesisState.prototype.clearParams = function() {
  return this.setParams(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.injective.tokenfactory.v1beta1.GenesisState.prototype.hasParams = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * repeated GenesisDenom factory_denoms = 2;
 * @return {!Array<!proto.injective.tokenfactory.v1beta1.GenesisDenom>}
 */
proto.injective.tokenfactory.v1beta1.GenesisState.prototype.getFactoryDenomsList = function() {
  return /** @type{!Array<!proto.injective.tokenfactory.v1beta1.GenesisDenom>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.injective.tokenfactory.v1beta1.GenesisDenom, 2));
};


/**
 * @param {!Array<!proto.injective.tokenfactory.v1beta1.GenesisDenom>} value
 * @return {!proto.injective.tokenfactory.v1beta1.GenesisState} returns this
*/
proto.injective.tokenfactory.v1beta1.GenesisState.prototype.setFactoryDenomsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.injective.tokenfactory.v1beta1.GenesisDenom=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.tokenfactory.v1beta1.GenesisDenom}
 */
proto.injective.tokenfactory.v1beta1.GenesisState.prototype.addFactoryDenoms = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.injective.tokenfactory.v1beta1.GenesisDenom, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.tokenfactory.v1beta1.GenesisState} returns this
 */
proto.injective.tokenfactory.v1beta1.GenesisState.prototype.clearFactoryDenomsList = function() {
  return this.setFactoryDenomsList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.tokenfactory.v1beta1.GenesisDenom.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.tokenfactory.v1beta1.GenesisDenom.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.tokenfactory.v1beta1.GenesisDenom} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.tokenfactory.v1beta1.GenesisDenom.toObject = function(includeInstance, msg) {
  var f, obj = {
    denom: jspb.Message.getFieldWithDefault(msg, 1, ""),
    authorityMetadata: (f = msg.getAuthorityMetadata()) && injective_tokenfactory_v1beta1_authorityMetadata_pb.DenomAuthorityMetadata.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.tokenfactory.v1beta1.GenesisDenom}
 */
proto.injective.tokenfactory.v1beta1.GenesisDenom.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.tokenfactory.v1beta1.GenesisDenom;
  return proto.injective.tokenfactory.v1beta1.GenesisDenom.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.tokenfactory.v1beta1.GenesisDenom} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.tokenfactory.v1beta1.GenesisDenom}
 */
proto.injective.tokenfactory.v1beta1.GenesisDenom.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setDenom(value);
      break;
    case 2:
      var value = new injective_tokenfactory_v1beta1_authorityMetadata_pb.DenomAuthorityMetadata;
      reader.readMessage(value,injective_tokenfactory_v1beta1_authorityMetadata_pb.DenomAuthorityMetadata.deserializeBinaryFromReader);
      msg.setAuthorityMetadata(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.tokenfactory.v1beta1.GenesisDenom.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.tokenfactory.v1beta1.GenesisDenom.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.tokenfactory.v1beta1.GenesisDenom} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.tokenfactory.v1beta1.GenesisDenom.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getDenom();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getAuthorityMetadata();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      injective_tokenfactory_v1beta1_authorityMetadata_pb.DenomAuthorityMetadata.serializeBinaryToWriter
    );
  }
};


/**
 * optional string denom = 1;
 * @return {string}
 */
proto.injective.tokenfactory.v1beta1.GenesisDenom.prototype.getDenom = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.tokenfactory.v1beta1.GenesisDenom} returns this
 */
proto.injective.tokenfactory.v1beta1.GenesisDenom.prototype.setDenom = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional DenomAuthorityMetadata authority_metadata = 2;
 * @return {?proto.injective.tokenfactory.v1beta1.DenomAuthorityMetadata}
 */
proto.injective.tokenfactory.v1beta1.GenesisDenom.prototype.getAuthorityMetadata = function() {
  return /** @type{?proto.injective.tokenfactory.v1beta1.DenomAuthorityMetadata} */ (
    jspb.Message.getWrapperField(this, injective_tokenfactory_v1beta1_authorityMetadata_pb.DenomAuthorityMetadata, 2));
};


/**
 * @param {?proto.injective.tokenfactory.v1beta1.DenomAuthorityMetadata|undefined} value
 * @return {!proto.injective.tokenfactory.v1beta1.GenesisDenom} returns this
*/
proto.injective.tokenfactory.v1beta1.GenesisDenom.prototype.setAuthorityMetadata = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.injective.tokenfactory.v1beta1.GenesisDenom} returns this
 */
proto.injective.tokenfactory.v1beta1.GenesisDenom.prototype.clearAuthorityMetadata = function() {
  return this.setAuthorityMetadata(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.injective.tokenfactory.v1beta1.GenesisDenom.prototype.hasAuthorityMetadata = function() {
  return jspb.Message.getField(this, 2) != null;
};


goog.object.extend(exports, proto.injective.tokenfactory.v1beta1);
