// @injectivelabs/chain-api/injective/tokenfactory/v1beta1/query_pb_service.js
// package: injective.tokenfactory.v1beta1
// file: injective/tokenfactory/v1beta1/query.proto

var injective_tokenfactory_v1beta1_query_pb = require("../../../injective/tokenfactory/v1beta1/query_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var Query = (function () {
  function Query() {}
  Query.serviceName = "injective.tokenfactory.v1beta1.Query";
  return Query;
}());

Query.Params = {
  methodName: "Params",
  service: Query,
  requestStream: false,
  responseStream: false,
  requestType: injective_tokenfactory_v1beta1_query_pb.QueryParamsRequest,
  responseType: injective_tokenfactory_v1beta1_query_pb.QueryParamsResponse
};

Query.DenomAuthorityMetadata = {
  methodName: "DenomAuthorityMetadata",
  service: Query,
  requestStream: false,
  responseStream: false,
  requestType: injective_tokenfactory_v1beta1_query_pb.QueryDenomAuthorityMetadataRequest,
  responseType: injective_tokenfactory_v1beta1_query_pb.QueryDenomAuthorityMetadataResponse
};

Query.DenomsFromCreator = {
  methodName: "DenomsFromCreator",
  service: Query,
  requestStream: false,
  responseStream: false,
  requestType: injective_tokenfactory_v1beta1_query_pb.QueryDenomsFromCreatorRequest,
  responseType: injective_tokenfactory_v1beta1_query_pb.QueryDenomsFromCreatorResponse
};

Query.TokenfactoryModuleState = {
  methodName: "TokenfactoryModuleState",
  service: Query,
  requestStream: false,
  responseStream: false,
  requestType: injective_tokenfactory_v1beta1_query_pb.QueryModuleStateRequest,
  responseType: injective_tokenfactory_v1beta1_query_pb.QueryModuleStateResponse
};

exports.Query = Query;

function QueryClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

QueryClient.prototype.params = function params(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Query.Params, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

QueryClient.prototype.denomAuthorityMetadata = function denomAuthorityMetadata(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Query.DenomAuthorityMetadata, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

QueryClient.prototype.denomsFromCreator = function denomsFromCreator(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Query.DenomsFromCreator, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

QueryClient.prototype.tokenfactoryModuleState = function tokenfactoryModuleState(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Query.TokenfactoryModuleState, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

exports.QueryClient = QueryClient;
