// @injectivelabs/chain-api/injective/tokenfactory/v1beta1/tx_pb_service.js
// package: injective.tokenfactory.v1beta1
// file: injective/tokenfactory/v1beta1/tx.proto

var injective_tokenfactory_v1beta1_tx_pb = require("../../../injective/tokenfactory/v1beta1/tx_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var Msg = (function () {
  function Msg() {}
  Msg.serviceName = "injective.tokenfactory.v1beta1.Msg";
  return Msg;
}());

Msg.CreateDenom = {
  methodName: "CreateDenom",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: injective_tokenfactory_v1beta1_tx_pb.MsgCreateDenom,
  responseType: injective_tokenfactory_v1beta1_tx_pb.MsgCreateDenomResponse
};

Msg.Mint = {
  methodName: "Mint",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: injective_tokenfactory_v1beta1_tx_pb.MsgMint,
  responseType: injective_tokenfactory_v1beta1_tx_pb.MsgMintResponse
};

Msg.Burn = {
  methodName: "Burn",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: injective_tokenfactory_v1beta1_tx_pb.MsgBurn,
  responseType: injective_tokenfactory_v1beta1_tx_pb.MsgBurnResponse
};

Msg.ChangeAdmin = {
  methodName: "ChangeAdmin",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: injective_tokenfactory_v1beta1_tx_pb.MsgChangeAdmin,
  responseType: injective_tokenfactory_v1beta1_tx_pb.MsgChangeAdminResponse
};

Msg.SetDenomMetadata = {
  methodName: "SetDenomMetadata",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: injective_tokenfactory_v1beta1_tx_pb.MsgSetDenomMetadata,
  responseType: injective_tokenfactory_v1beta1_tx_pb.MsgSetDenomMetadataResponse
};

exports.Msg = Msg;

function MsgClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

MsgClient.prototype.createDenom = function createDenom(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.CreateDenom, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.mint = function mint(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.Mint, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.burn = function burn(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.Burn, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.changeAdmin = function changeAdmin(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.ChangeAdmin, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.setDenomMetadata = function setDenomMetadata(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.SetDenomMetadata, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

exports.MsgClient = MsgClient;
