// @injectivelabs/chain-api/injective/wasmx/v1/genesis_pb.js
// source: injective/wasmx/v1/genesis.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() {
  if (this) { return this; }
  if (typeof window !== 'undefined') { return window; }
  if (typeof global !== 'undefined') { return global; }
  if (typeof self !== 'undefined') { return self; }
  return Function('return this')();
}.call(null));

var injective_wasmx_v1_wasmx_pb = require('../../../injective/wasmx/v1/wasmx_pb.js');
goog.object.extend(proto, injective_wasmx_v1_wasmx_pb);
var gogoproto_gogo_pb = require('../../../gogoproto/gogo_pb.js');
goog.object.extend(proto, gogoproto_gogo_pb);
goog.exportSymbol('proto.injective.wasmx.v1.GenesisState', null, global);
goog.exportSymbol('proto.injective.wasmx.v1.RegisteredContractWithAddress', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.wasmx.v1.RegisteredContractWithAddress = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.injective.wasmx.v1.RegisteredContractWithAddress, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.wasmx.v1.RegisteredContractWithAddress.displayName = 'proto.injective.wasmx.v1.RegisteredContractWithAddress';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.injective.wasmx.v1.GenesisState = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.injective.wasmx.v1.GenesisState.repeatedFields_, null);
};
goog.inherits(proto.injective.wasmx.v1.GenesisState, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.injective.wasmx.v1.GenesisState.displayName = 'proto.injective.wasmx.v1.GenesisState';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.wasmx.v1.RegisteredContractWithAddress.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.wasmx.v1.RegisteredContractWithAddress.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.wasmx.v1.RegisteredContractWithAddress} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.wasmx.v1.RegisteredContractWithAddress.toObject = function(includeInstance, msg) {
  var f, obj = {
    address: jspb.Message.getFieldWithDefault(msg, 1, ""),
    registeredContract: (f = msg.getRegisteredContract()) && injective_wasmx_v1_wasmx_pb.RegisteredContract.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.wasmx.v1.RegisteredContractWithAddress}
 */
proto.injective.wasmx.v1.RegisteredContractWithAddress.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.wasmx.v1.RegisteredContractWithAddress;
  return proto.injective.wasmx.v1.RegisteredContractWithAddress.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.wasmx.v1.RegisteredContractWithAddress} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.wasmx.v1.RegisteredContractWithAddress}
 */
proto.injective.wasmx.v1.RegisteredContractWithAddress.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setAddress(value);
      break;
    case 2:
      var value = new injective_wasmx_v1_wasmx_pb.RegisteredContract;
      reader.readMessage(value,injective_wasmx_v1_wasmx_pb.RegisteredContract.deserializeBinaryFromReader);
      msg.setRegisteredContract(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.wasmx.v1.RegisteredContractWithAddress.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.wasmx.v1.RegisteredContractWithAddress.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.wasmx.v1.RegisteredContractWithAddress} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.wasmx.v1.RegisteredContractWithAddress.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getAddress();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getRegisteredContract();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      injective_wasmx_v1_wasmx_pb.RegisteredContract.serializeBinaryToWriter
    );
  }
};


/**
 * optional string address = 1;
 * @return {string}
 */
proto.injective.wasmx.v1.RegisteredContractWithAddress.prototype.getAddress = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.injective.wasmx.v1.RegisteredContractWithAddress} returns this
 */
proto.injective.wasmx.v1.RegisteredContractWithAddress.prototype.setAddress = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional RegisteredContract registered_contract = 2;
 * @return {?proto.injective.wasmx.v1.RegisteredContract}
 */
proto.injective.wasmx.v1.RegisteredContractWithAddress.prototype.getRegisteredContract = function() {
  return /** @type{?proto.injective.wasmx.v1.RegisteredContract} */ (
    jspb.Message.getWrapperField(this, injective_wasmx_v1_wasmx_pb.RegisteredContract, 2));
};


/**
 * @param {?proto.injective.wasmx.v1.RegisteredContract|undefined} value
 * @return {!proto.injective.wasmx.v1.RegisteredContractWithAddress} returns this
*/
proto.injective.wasmx.v1.RegisteredContractWithAddress.prototype.setRegisteredContract = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.injective.wasmx.v1.RegisteredContractWithAddress} returns this
 */
proto.injective.wasmx.v1.RegisteredContractWithAddress.prototype.clearRegisteredContract = function() {
  return this.setRegisteredContract(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.injective.wasmx.v1.RegisteredContractWithAddress.prototype.hasRegisteredContract = function() {
  return jspb.Message.getField(this, 2) != null;
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.injective.wasmx.v1.GenesisState.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.injective.wasmx.v1.GenesisState.prototype.toObject = function(opt_includeInstance) {
  return proto.injective.wasmx.v1.GenesisState.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.injective.wasmx.v1.GenesisState} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.wasmx.v1.GenesisState.toObject = function(includeInstance, msg) {
  var f, obj = {
    params: (f = msg.getParams()) && injective_wasmx_v1_wasmx_pb.Params.toObject(includeInstance, f),
    registeredContractsList: jspb.Message.toObjectList(msg.getRegisteredContractsList(),
    proto.injective.wasmx.v1.RegisteredContractWithAddress.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.injective.wasmx.v1.GenesisState}
 */
proto.injective.wasmx.v1.GenesisState.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.injective.wasmx.v1.GenesisState;
  return proto.injective.wasmx.v1.GenesisState.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.injective.wasmx.v1.GenesisState} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.injective.wasmx.v1.GenesisState}
 */
proto.injective.wasmx.v1.GenesisState.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new injective_wasmx_v1_wasmx_pb.Params;
      reader.readMessage(value,injective_wasmx_v1_wasmx_pb.Params.deserializeBinaryFromReader);
      msg.setParams(value);
      break;
    case 2:
      var value = new proto.injective.wasmx.v1.RegisteredContractWithAddress;
      reader.readMessage(value,proto.injective.wasmx.v1.RegisteredContractWithAddress.deserializeBinaryFromReader);
      msg.addRegisteredContracts(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.injective.wasmx.v1.GenesisState.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.injective.wasmx.v1.GenesisState.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.injective.wasmx.v1.GenesisState} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.injective.wasmx.v1.GenesisState.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getParams();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      injective_wasmx_v1_wasmx_pb.Params.serializeBinaryToWriter
    );
  }
  f = message.getRegisteredContractsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      proto.injective.wasmx.v1.RegisteredContractWithAddress.serializeBinaryToWriter
    );
  }
};


/**
 * optional Params params = 1;
 * @return {?proto.injective.wasmx.v1.Params}
 */
proto.injective.wasmx.v1.GenesisState.prototype.getParams = function() {
  return /** @type{?proto.injective.wasmx.v1.Params} */ (
    jspb.Message.getWrapperField(this, injective_wasmx_v1_wasmx_pb.Params, 1));
};


/**
 * @param {?proto.injective.wasmx.v1.Params|undefined} value
 * @return {!proto.injective.wasmx.v1.GenesisState} returns this
*/
proto.injective.wasmx.v1.GenesisState.prototype.setParams = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.injective.wasmx.v1.GenesisState} returns this
 */
proto.injective.wasmx.v1.GenesisState.prototype.clearParams = function() {
  return this.setParams(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.injective.wasmx.v1.GenesisState.prototype.hasParams = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * repeated RegisteredContractWithAddress registered_contracts = 2;
 * @return {!Array<!proto.injective.wasmx.v1.RegisteredContractWithAddress>}
 */
proto.injective.wasmx.v1.GenesisState.prototype.getRegisteredContractsList = function() {
  return /** @type{!Array<!proto.injective.wasmx.v1.RegisteredContractWithAddress>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.injective.wasmx.v1.RegisteredContractWithAddress, 2));
};


/**
 * @param {!Array<!proto.injective.wasmx.v1.RegisteredContractWithAddress>} value
 * @return {!proto.injective.wasmx.v1.GenesisState} returns this
*/
proto.injective.wasmx.v1.GenesisState.prototype.setRegisteredContractsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.injective.wasmx.v1.RegisteredContractWithAddress=} opt_value
 * @param {number=} opt_index
 * @return {!proto.injective.wasmx.v1.RegisteredContractWithAddress}
 */
proto.injective.wasmx.v1.GenesisState.prototype.addRegisteredContracts = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.injective.wasmx.v1.RegisteredContractWithAddress, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.injective.wasmx.v1.GenesisState} returns this
 */
proto.injective.wasmx.v1.GenesisState.prototype.clearRegisteredContractsList = function() {
  return this.setRegisteredContractsList([]);
};


goog.object.extend(exports, proto.injective.wasmx.v1);
