// @injectivelabs/chain-api/injective/wasmx/v1/query_pb_service.js
// package: injective.wasmx.v1
// file: injective/wasmx/v1/query.proto

var injective_wasmx_v1_query_pb = require("../../../injective/wasmx/v1/query_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var Query = (function () {
  function Query() {}
  Query.serviceName = "injective.wasmx.v1.Query";
  return Query;
}());

Query.WasmxParams = {
  methodName: "WasmxParams",
  service: Query,
  requestStream: false,
  responseStream: false,
  requestType: injective_wasmx_v1_query_pb.QueryWasmxParamsRequest,
  responseType: injective_wasmx_v1_query_pb.QueryWasmxParamsResponse
};

Query.WasmxModuleState = {
  methodName: "WasmxModuleState",
  service: Query,
  requestStream: false,
  responseStream: false,
  requestType: injective_wasmx_v1_query_pb.QueryModuleStateRequest,
  responseType: injective_wasmx_v1_query_pb.QueryModuleStateResponse
};

exports.Query = Query;

function QueryClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

QueryClient.prototype.wasmxParams = function wasmxParams(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Query.WasmxParams, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

QueryClient.prototype.wasmxModuleState = function wasmxModuleState(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Query.WasmxModuleState, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

exports.QueryClient = QueryClient;
