// @injectivelabs/chain-api/injective/wasmx/v1/tx_pb_service.js
// package: injective.wasmx.v1beta1
// file: injective/wasmx/v1/tx.proto

var injective_wasmx_v1_tx_pb = require("../../../injective/wasmx/v1/tx_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var Msg = (function () {
  function Msg() {}
  Msg.serviceName = "injective.wasmx.v1beta1.Msg";
  return Msg;
}());

Msg.UpdateRegistryContractParams = {
  methodName: "UpdateRegistryContractParams",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: injective_wasmx_v1_tx_pb.MsgUpdateContract,
  responseType: injective_wasmx_v1_tx_pb.MsgUpdateContractResponse
};

Msg.ActivateRegistryContract = {
  methodName: "ActivateRegistryContract",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: injective_wasmx_v1_tx_pb.MsgActivateContract,
  responseType: injective_wasmx_v1_tx_pb.MsgActivateContractResponse
};

Msg.DeactivateRegistryContract = {
  methodName: "DeactivateRegistryContract",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: injective_wasmx_v1_tx_pb.MsgDeactivateContract,
  responseType: injective_wasmx_v1_tx_pb.MsgDeactivateContractResponse
};

Msg.Transfer = {
  methodName: "Transfer",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: injective_wasmx_v1_tx_pb.MsgTransfer,
  responseType: injective_wasmx_v1_tx_pb.MsgTransferResponse
};

exports.Msg = Msg;

function MsgClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

MsgClient.prototype.updateRegistryContractParams = function updateRegistryContractParams(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.UpdateRegistryContractParams, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.activateRegistryContract = function activateRegistryContract(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.ActivateRegistryContract, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.deactivateRegistryContract = function deactivateRegistryContract(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.DeactivateRegistryContract, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.transfer = function transfer(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.Transfer, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

exports.MsgClient = MsgClient;
