// @injectivelabs/chain-api/injective/ocr/v1beta1/query_pb_service.js
// package: injective.ocr.v1beta1
// file: injective/ocr/v1beta1/query.proto

var injective_ocr_v1beta1_query_pb = require("../../../injective/ocr/v1beta1/query_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var Query = (function () {
  function Query() {}
  Query.serviceName = "injective.ocr.v1beta1.Query";
  return Query;
}());

Query.Params = {
  methodName: "Params",
  service: Query,
  requestStream: false,
  responseStream: false,
  requestType: injective_ocr_v1beta1_query_pb.QueryParamsRequest,
  responseType: injective_ocr_v1beta1_query_pb.QueryParamsResponse
};

Query.FeedConfig = {
  methodName: "FeedConfig",
  service: Query,
  requestStream: false,
  responseStream: false,
  requestType: injective_ocr_v1beta1_query_pb.QueryFeedConfigRequest,
  responseType: injective_ocr_v1beta1_query_pb.QueryFeedConfigResponse
};

Query.FeedConfigInfo = {
  methodName: "FeedConfigInfo",
  service: Query,
  requestStream: false,
  responseStream: false,
  requestType: injective_ocr_v1beta1_query_pb.QueryFeedConfigInfoRequest,
  responseType: injective_ocr_v1beta1_query_pb.QueryFeedConfigInfoResponse
};

Query.LatestRound = {
  methodName: "LatestRound",
  service: Query,
  requestStream: false,
  responseStream: false,
  requestType: injective_ocr_v1beta1_query_pb.QueryLatestRoundRequest,
  responseType: injective_ocr_v1beta1_query_pb.QueryLatestRoundResponse
};

Query.LatestTransmissionDetails = {
  methodName: "LatestTransmissionDetails",
  service: Query,
  requestStream: false,
  responseStream: false,
  requestType: injective_ocr_v1beta1_query_pb.QueryLatestTransmissionDetailsRequest,
  responseType: injective_ocr_v1beta1_query_pb.QueryLatestTransmissionDetailsResponse
};

Query.OwedAmount = {
  methodName: "OwedAmount",
  service: Query,
  requestStream: false,
  responseStream: false,
  requestType: injective_ocr_v1beta1_query_pb.QueryOwedAmountRequest,
  responseType: injective_ocr_v1beta1_query_pb.QueryOwedAmountResponse
};

Query.OcrModuleState = {
  methodName: "OcrModuleState",
  service: Query,
  requestStream: false,
  responseStream: false,
  requestType: injective_ocr_v1beta1_query_pb.QueryModuleStateRequest,
  responseType: injective_ocr_v1beta1_query_pb.QueryModuleStateResponse
};

exports.Query = Query;

function QueryClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

QueryClient.prototype.params = function params(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Query.Params, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

QueryClient.prototype.feedConfig = function feedConfig(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Query.FeedConfig, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

QueryClient.prototype.feedConfigInfo = function feedConfigInfo(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Query.FeedConfigInfo, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

QueryClient.prototype.latestRound = function latestRound(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Query.LatestRound, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

QueryClient.prototype.latestTransmissionDetails = function latestTransmissionDetails(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Query.LatestTransmissionDetails, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

QueryClient.prototype.owedAmount = function owedAmount(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Query.OwedAmount, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

QueryClient.prototype.ocrModuleState = function ocrModuleState(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Query.OcrModuleState, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

exports.QueryClient = QueryClient;
