// @injectivelabs/chain-api/injective/ocr/v1beta1/tx_pb_service.js
// package: injective.ocr.v1beta1
// file: injective/ocr/v1beta1/tx.proto

var injective_ocr_v1beta1_tx_pb = require("../../../injective/ocr/v1beta1/tx_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var Msg = (function () {
  function Msg() {}
  Msg.serviceName = "injective.ocr.v1beta1.Msg";
  return Msg;
}());

Msg.CreateFeed = {
  methodName: "CreateFeed",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: injective_ocr_v1beta1_tx_pb.MsgCreateFeed,
  responseType: injective_ocr_v1beta1_tx_pb.MsgCreateFeedResponse
};

Msg.UpdateFeed = {
  methodName: "UpdateFeed",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: injective_ocr_v1beta1_tx_pb.MsgUpdateFeed,
  responseType: injective_ocr_v1beta1_tx_pb.MsgUpdateFeedResponse
};

Msg.Transmit = {
  methodName: "Transmit",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: injective_ocr_v1beta1_tx_pb.MsgTransmit,
  responseType: injective_ocr_v1beta1_tx_pb.MsgTransmitResponse
};

Msg.FundFeedRewardPool = {
  methodName: "FundFeedRewardPool",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: injective_ocr_v1beta1_tx_pb.MsgFundFeedRewardPool,
  responseType: injective_ocr_v1beta1_tx_pb.MsgFundFeedRewardPoolResponse
};

Msg.WithdrawFeedRewardPool = {
  methodName: "WithdrawFeedRewardPool",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: injective_ocr_v1beta1_tx_pb.MsgWithdrawFeedRewardPool,
  responseType: injective_ocr_v1beta1_tx_pb.MsgWithdrawFeedRewardPoolResponse
};

Msg.SetPayees = {
  methodName: "SetPayees",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: injective_ocr_v1beta1_tx_pb.MsgSetPayees,
  responseType: injective_ocr_v1beta1_tx_pb.MsgSetPayeesResponse
};

Msg.TransferPayeeship = {
  methodName: "TransferPayeeship",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: injective_ocr_v1beta1_tx_pb.MsgTransferPayeeship,
  responseType: injective_ocr_v1beta1_tx_pb.MsgTransferPayeeshipResponse
};

Msg.AcceptPayeeship = {
  methodName: "AcceptPayeeship",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: injective_ocr_v1beta1_tx_pb.MsgAcceptPayeeship,
  responseType: injective_ocr_v1beta1_tx_pb.MsgAcceptPayeeshipResponse
};

exports.Msg = Msg;

function MsgClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

MsgClient.prototype.createFeed = function createFeed(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.CreateFeed, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.updateFeed = function updateFeed(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.UpdateFeed, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.transmit = function transmit(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.Transmit, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.fundFeedRewardPool = function fundFeedRewardPool(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.FundFeedRewardPool, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.withdrawFeedRewardPool = function withdrawFeedRewardPool(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.WithdrawFeedRewardPool, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.setPayees = function setPayees(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.SetPayees, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.transferPayeeship = function transferPayeeship(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.TransferPayeeship, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.acceptPayeeship = function acceptPayeeship(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.AcceptPayeeship, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

exports.MsgClient = MsgClient;
